diff --git a/Makefile b/Makefile
index fc0c009..6d5ebc0 100644
--- a/Makefile
+++ b/Makefile
@@ -41,7 +41,7 @@ clean_htslib:
 
 %.o: src/%.c
 	mkdir -p obj && \
-		$(CC) -c -pthread -Wall -fstack-protector-strong -D_FORTIFY_SOURCE=2 $(CFLAGS) \
+		$(CC) -c -Wall -fstack-protector-strong -D_FORTIFY_SOURCE=2 $(CFLAGS) \
 		-Isrc -Ihtslib $(EXTRA_CFLAGS) $^ -o $@
 
 .PHONY: clean_obj
@@ -50,7 +50,7 @@ clean_obj:
 
 
 modbam2bed: modbam2bed.o common.o counts.o bamiter.o args.o $(STATIC_HTSLIB)
-	$(CC) -pthread -Wall -fstack-protector-strong -D_FORTIFY_SOURCE=2 $(CFLAGS) \
+	$(CC) -Wall -fstack-protector-strong -D_FORTIFY_SOURCE=2 $(CFLAGS) \
 		-Isrc -Ihtslib $(EXTRA_CFLAGS) $(EXTRA_LDFLAGS)\
 		$^ $(ARGP) \
 		-lm -lz -llzma -lbz2 -lpthread -lcurl -lcrypto $(EXTRA_LIBS) \
