diff --git a/joblib/backports.py b/joblib/backports.py
index 7dd3df1..a49ef78 100644
--- a/joblib/backports.py
+++ b/joblib/backports.py
@@ -3,7 +3,6 @@ Backports of fixes for joblib dependencies
 """
 import os
 import time
-import ctypes
 import sys
 
 from distutils.version import LooseVersion
@@ -34,6 +33,8 @@ except ImportError:
 
 if os.name == 'nt':
     error_access_denied = 5
+    import ctypes
+
     try:
         from os import replace
     except ImportError:
