diff --git a/joblib/backports.py b/joblib/backports.py
index 7dd3df1..a49ef78 100644
--- a/joblib/backports.py
+++ b/joblib/backports.py
@@ -3,7 +3,6 @@ Backports of fixes for joblib dependencies
 """
 import os
 import time
-import ctypes
 import sys

 from pkg_resources import parse_version
@@ -33,6 +32,7 @@


 if os.name == 'nt':
+    import ctypes
     # https://github.com/joblib/joblib/issues/540
     access_denied_errors = (5, 13)
     try:
