From 0a6cc79e6814d7e2dbc637541cad86cebc47f855 Mon Sep 17 00:00:00 2001
From: Hanno Rein <hanno@hanno-rein.de>
Date: Wed, 6 Nov 2024 17:37:22 -0500
Subject: [PATCH 3/3] install_name

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 6443cb27..965b2fc9 100644
--- a/setup.py
+++ b/setup.py
@@ -23,8 +23,8 @@ except:
 extra_link_args=[]
 if sys.platform == 'darwin':
     config_vars = sysconfig.get_config_vars()
-    config_vars['LDSHARED'] = config_vars['LDSHARED'].replace('-bundle', '-shared')
-    extra_link_args=['-Wl,-install_name,@rpath/librebound'+suffix]
+    #config_vars['LDSHARED'] = config_vars['LDSHARED'].replace('-bundle', '-shared')
+    #extra_link_args=['-Wl,-install_name,@rpath/librebound'+suffix]
 if sys.platform == 'win32':
     extra_compile_args=[ghash_arg, '-DLIBREBOUND', '-D_GNU_SOURCE', '-DSERVER']
 else:
-- 
2.39.5 (Apple Git-154)

