From 2bbbc27bf5f7358ecc030a2cc0e6a9785a25b6f5 Mon Sep 17 00:00:00 2001
From: Jeffrey Bouas <ignirtoq@gmail.com>
Date: Mon, 27 Aug 2018 11:39:28 -0500
Subject: [PATCH 05/17] Tweak default value of economy parameter and update
 docstring.

---
 sparseqr/sparseqr.py | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/sparseqr/sparseqr.py b/sparseqr/sparseqr.py
index 401270f..fe2b677 100644
--- a/sparseqr/sparseqr.py
+++ b/sparseqr/sparseqr.py
@@ -200,7 +200,7 @@ def cholmod_free_dense( A ):
 
 ## Solvers
 
-def qr( A, tolerance = None, economy = False ):
+def qr( A, tolerance = None, economy = None ):
     '''
     Given a sparse matrix A,
     returns Q, R, E, rank such that:
@@ -211,6 +211,11 @@ def qr( A, tolerance = None, economy = False ):
         #define SPQR_DEFAULT_TOL ...       /* if tol <= -2, the default tol is used */
         #define SPQR_NO_TOL ...            /* if -2 < tol < 0, then no tol is used */
 
+    For A an m-by-n matrix, Q will be m-by-m and R will be m-by-n.
+
+    If optional `economy` parameter is truthy, Q will be m-by-k and R will be
+    k-by-n, where k = min(m, n).
+
     The performance-optimal format for A is scipy.sparse.coo_matrix.
 
     For solving systems of the form A x = b, see solve().
@@ -254,6 +259,8 @@ def qr( A, tolerance = None, economy = False ):
 
     if tolerance is None: tolerance = lib.SPQR_DEFAULT_TOL
 
+    if economy is None: economy = False
+
     econ = A.shape[1] if economy else A.shape[0]
 
     rank = lib.SuiteSparseQR_C_QR(
-- 
2.25.1

