From b4eda5e00ff1d7206ff8d51920b43d06f9e4fa66 Mon Sep 17 00:00:00 2001
From: Yotam Gingold <yotam@yotamgingold.com>
Date: Mon, 9 Aug 2021 04:27:10 -0400
Subject: [PATCH 12/17] build module at the right place, take 2

---
 sparseqr/sparseqr_gen.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sparseqr/sparseqr_gen.py b/sparseqr/sparseqr_gen.py
index fe14806..bd69812 100644
--- a/sparseqr/sparseqr_gen.py
+++ b/sparseqr/sparseqr_gen.py
@@ -367,7 +367,7 @@ cholmod_sparse *SuiteSparseQR_C_backslash_sparse   /* returns X, or NULL */
 """)
 
 def main():
-    ffibuilder.compile( verbose = True, tmpdir = os.path.dirname(__file__) )
+    ffibuilder.compile( verbose = True, tmpdir = os.path.dirname(os.path.dirname(os.path.abspath(__file__))) )
 
 if __name__ == "__main__":
     main()
-- 
2.25.1

