From cab81905eca1c75a0ccdf39c6fd694320a898e44 Mon Sep 17 00:00:00 2001
From: giuseppe <capassog97@gmail.com>
Date: Sat, 28 Dec 2024 16:37:12 +0100
Subject: [PATCH 2/2] Added UnsupportedParamsError

---
 src/argon2/exceptions.py | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/argon2/exceptions.py b/src/argon2/exceptions.py
index c81b4b9..53ddae5 100644
--- a/src/argon2/exceptions.py
+++ b/src/argon2/exceptions.py
@@ -46,6 +46,14 @@ class InvalidHashError(ValueError):
     """
 
 
+class UnsupportedParamsError(ValueError):
+    """
+    Raised if the current platform doesn not support the parameters.
+
+    Eg. In Wasm32, parallelism must be set to 1.
+    """
+
+
 InvalidHash = InvalidHashError
 """
 Deprecated alias for :class:`InvalidHashError`.
-- 
2.47.1

