From e32b9c86d168177c3e47194c837a263aa931cd0f Mon Sep 17 00:00:00 2001
From: Jo Bovy <bovy@astro.utoronto.ca>
Date: Wed, 8 Jun 2022 15:57:49 -0400
Subject: [PATCH 2/2] Don't use OpenMP for pyodide build

---
 setup.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index b94620b..035d984 100644
--- a/setup.py
+++ b/setup.py
@@ -61,7 +61,11 @@ if use_intel_compiler: # OpenMP by default included for Intel, see #416
 try:
     openmp_pos = sys.argv.index('--no-openmp')
 except ValueError:
-    extra_compile_args = ["-fopenmp" if not WIN32 else "/openmp"]
+    if "PYODIDE" in os.environ:
+        extra_compile_args= ["-DNO_OMP"]
+        galpy_c_libraries.remove('gomp')
+    else:
+        extra_compile_args = ["-fopenmp" if not WIN32 else "/openmp"]
 else:
     del sys.argv[openmp_pos]
     extra_compile_args= ["-DNO_OMP"]
-- 
2.32.1 (Apple Git-133)

