From b23a471e5c5cbd0fa8229a3af676e148e394b2d8 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Wed, 31 Jan 2024 15:36:19 -0800
Subject: [PATCH] Unpin setuptools

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 0017636..f977f3f 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -12,5 +12,5 @@ requires = [
   "numpy==1.22.2; python_version>='3.11'",
   "pip",
   "scikit-build>=0.14.0",
-  "setuptools==59.2.0",
+  "setuptools",
 ]
-- 
2.25.1

