From 21dd8d3081d792b36d617220218b413ace662205 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Sun, 26 Mar 2023 18:57:02 -0500
Subject: [PATCH 1/2] Allow cross building packages

---
 bin/BuildPackages.sh | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/bin/BuildPackages.sh b/bin/BuildPackages.sh
index fb17bec8..c1055130 100755
--- a/bin/BuildPackages.sh
+++ b/bin/BuildPackages.sh
@@ -194,11 +194,7 @@ run_configure_and_make() {
   then
     if grep Autoconf ./configure > /dev/null
     then
-      local PKG_NAME=$($GAP -q -T -A -M <<GAPInput
-Read("PackageInfo.g");
-Print(GAPInfo.PackageInfoCurrent.PackageName);
-GAPInput
-)
+      local PKG_NAME=$(cat ./PackageInfo.g | grep "PackageName := " | cut -b 17- | rev | cut -b 3- | rev)
       local CONFIG_ARGS_FLAG_NAME="PACKAGE_CONFIG_ARGS_${PKG_NAME}"
       echo_run ./configure --with-gaproot="$GAPROOT" $CONFIGFLAGS ${!CONFIG_ARGS_FLAG_NAME}
       echo_run "$MAKE" clean
-- 
2.37.3

