From 7a0d3099be580cdb10357862d07133c296224e78 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Wed, 20 Mar 2024 18:41:24 +0100
Subject: [PATCH] Remove emscripten shim

---
 src/urllib3/__init__.py | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/src/urllib3/__init__.py b/src/urllib3/__init__.py
index 1e0bf37b..be84504e 100644
--- a/src/urllib3/__init__.py
+++ b/src/urllib3/__init__.py
@@ -203,9 +203,3 @@ def request(
         timeout=timeout,
         json=json,
     )
-
-
-if sys.platform == "emscripten":
-    from .contrib.emscripten import inject_into_urllib3  # noqa: 401
-
-    inject_into_urllib3()
-- 
2.34.1

