--- gsw-3.4.0.orig/pyproject.toml	2020-08-24 13:43:49.000000000 -0300
+++ gsw-3.4.0/pyproject.toml	2022-05-05 10:25:07.125980604 -0300
@@ -1,3 +1,3 @@
 [build-system]
-requires = ["pip >9.0.1", "setuptools>=42", "wheel", "setuptools_scm[toml]>=3.4", "cython", "numpy"]
+requires = ["pip>9.0.1", "setuptools>=42", "wheel", "setuptools_scm[toml]>=3.4", "Cython", "numpy"]
 build-backend = "setuptools.build_meta"
