From aa6fd1737e1beedcf4f3985a2f8f2243ee23fa7b Mon Sep 17 00:00:00 2001
From: Hood <hood@mit.edu>
Date: Sat, 13 Nov 2021 14:10:54 -0800
Subject: [PATCH] Fix pushes_fd method signature

---
 src/encode.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/encode.c b/src/encode.c
index daa806ff4..d7498c25e 100644
--- a/src/encode.c
+++ b/src/encode.c
@@ -299,7 +299,7 @@ _setfd(ImagingEncoderObject *encoder, PyObject *args) {
 }
 
 static PyObject *
-_get_pushes_fd(ImagingEncoderObject *encoder) {
+_get_pushes_fd(ImagingEncoderObject *encoder, void* closure) {
     return PyBool_FromLong(encoder->pushes_fd);
 }
 
@@ -315,7 +315,7 @@ static struct PyMethodDef methods[] = {
 
 static struct PyGetSetDef getseters[] = {
     {"pushes_fd",
-     (getter)_get_pushes_fd,
+     _get_pushes_fd,
      NULL,
      "True if this decoder expects to push directly to self.fd",
      NULL},
-- 
2.17.1

