From 091eb5de67cd7d242d9d368e33b05ccc84aa8e56 Mon Sep 17 00:00:00 2001
From: Tamas Nepusz <ntamas@gmail.com>
Date: Tue, 11 Jul 2023 13:05:04 +0200
Subject: [PATCH 3/3] fix: explicitly turn off libxml2 in wasm builds

---
 setup.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup.py b/setup.py
index 36002acd..21872239 100644
--- a/setup.py
+++ b/setup.py
@@ -259,6 +259,7 @@ class IgraphCCoreCMakeBuilder:
                 return False
             args.insert(0, emcmake)
             args.append("-DIGRAPH_WARNINGS_AS_ERRORS:BOOL=OFF")
+            args.append("-DIGRAPH_GRAPHML_SUPPORT:BOOL=OFF")
 
         # Build the Python interface with vendored libraries
         for deps in "ARPACK BLAS GLPK GMP LAPACK".split():
-- 
2.41.0

