diff --git a/setup_build.py b/setup_build.py
index 6b66789..448774a 100644
--- a/setup_build.py
+++ b/setup_build.py
@@ -104,7 +104,7 @@ class h5py_build_ext(build_ext):
             settings['include_dirs'] += [mpi4py.get_include()]
 
         # TODO: should this only be done on UNIX?
-        if os.name != 'nt':
+        if False:
             settings['runtime_library_dirs'] = settings['library_dirs']
 
         def make_extension(module):
