From 2d6b0f7986afabcf17b049a199b0af795faf5db2 Mon Sep 17 00:00:00 2001
From: Hood <hood@mit.edu>
Date: Thu, 24 Jun 2021 04:08:02 -0700
Subject: [PATCH] Dummy out failing terser call

---
 emcc.py | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/emsdk/upstream/emscripten/eemcc.py b/emsdk/upstream/emscripten/emcc.py
index 839f791b3..b3eef03ef 100755
--- a/emsdk/upstream/emscripten/eemcc.py
+++ b/emsdk/upstream/emscripten/eemcc.py
@@ -2901,11 +2901,11 @@ def do_binaryen(target, options, wasm_target):
       # Closure can print out readable error messages (Closure will then
       # minify whitespace afterwards)
       save_intermediate_with_wasm('preclean', wasm_target)
-      final_js = building.minify_wasm_js(js_file=final_js,
-                                         wasm_file=wasm_target,
-                                         expensive_optimizations=will_metadce(),
-                                         minify_whitespace=minify_whitespace() and not options.use_closure_compiler,
-                                         debug_info=intermediate_debug_info)
+      # final_js = building.minify_wasm_js(js_file=final_js,
+      #                                    wasm_file=wasm_target,
+      #                                    expensive_optimizations=will_metadce(),
+      #                                    minify_whitespace=minify_whitespace() and not options.use_closure_compiler,
+      #                                    debug_info=intermediate_debug_info)
       save_intermediate_with_wasm('postclean', wasm_target)
 
   if shared.Settings.ASYNCIFY_LAZY_LOAD_CODE:
-- 
2.17.1

