From e83a295f8e1b8c48d4748d1811a4b22840f25e14 Mon Sep 17 00:00:00 2001
From: Hood <hood@mit.edu>
Date: Thu, 24 Jun 2021 04:08:02 -0700
Subject: [PATCH 1/4] Throw away errors in minify_wasm_js

---
 emcc.py | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/emcc.py b/emcc.py
index 78801d2d5..ada67e93f 100755
--- a/emcc.py
+++ b/emcc.py
@@ -3130,11 +3130,14 @@ def phase_binaryen(target, options, wasm_target):
       # Closure can print out readable error messages (Closure will then
       # minify whitespace afterwards)
       save_intermediate_with_wasm('preclean', wasm_target)
-      final_js = building.minify_wasm_js(js_file=final_js,
-                                         wasm_file=wasm_target,
-                                         expensive_optimizations=will_metadce(),
-                                         minify_whitespace=minify_whitespace() and not options.use_closure_compiler,
-                                         debug_info=intermediate_debug_info)
+      try:
+        final_js = building.minify_wasm_js(js_file=final_js,
+                                          wasm_file=wasm_target,
+                                          expensive_optimizations=will_metadce(),
+                                          minify_whitespace=minify_whitespace() and not options.use_closure_compiler,
+                                          debug_info=intermediate_debug_info)
+      except:
+        pass
       save_intermediate_with_wasm('postclean', wasm_target)
 
   if settings.ASYNCIFY_LAZY_LOAD_CODE:
-- 
2.25.1

