From 6d0a0edcc027668718bccfa12a08b6f793e83236 Mon Sep 17 00:00:00 2001
From: ryanking13 <def6488@gmail.com>
Date: Wed, 28 Jan 2026 11:01:56 +0900
Subject: [PATCH 3/3] Fix incorrect version number

---
 emscripten-version.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/emscripten-version.txt b/emscripten-version.txt
index 8f0a5940b..193541c80 100644
--- a/emscripten-version.txt
+++ b/emscripten-version.txt
@@ -1 +1 @@
-4.0.24-git
+5.0.0-git
-- 
2.29.2.windows.2

