From c6049eb3cf7cd6d3a112e2d0956977d00aa055ae Mon Sep 17 00:00:00 2001
From: ryanking13 <def6488@gmail.com>
Date: Sun, 22 Mar 2026 00:56:59 +0900
Subject: [PATCH 1/1] Allow passing --support-longjmp flag to embuilder

---
 embuilder.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/embuilder.py b/embuilder.py
index 6232fc5da..397c25426 100755
--- a/embuilder.py
+++ b/embuilder.py
@@ -204,6 +204,7 @@ def main():
   parser.add_argument('--lto=thin', dest='lto', action='store_const', const='thin', help='build bitcode object for ThinLTO')
   parser.add_argument('--pic', action='store_true',
                       help='build relocatable objects suitable for dynamic linking')
+  parser.add_argument('--support-longjmp', choices=['wasm', 'emscripten'], help='configure support for longjmp in wasm')
   parser.add_argument('-f', '--force', action='store_true',
                       help='force rebuild of target (by removing it first)')
   parser.add_argument('-v', '--verbose', action='store_true',
@@ -245,6 +246,9 @@ def main():
 
   if args.wasm64:
     settings.MEMORY64 = 1
+  
+  if args.support_longjmp:
+    settings.SUPPORT_LONGJMP = args.support_longjmp
 
   do_build = args.operation == 'build'
   do_clear = args.operation == 'clear'
-- 
2.43.0

