From 0c1ff8c2e704c39b5a2d8cf17e390b2adaaa28b6 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Mon, 20 Jul 2026 12:07:39 +0200
Subject: [PATCH 5/5] Turn FAKE_DYNLIBS back on

---
 src/settings.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/settings.js b/src/settings.js
index c9e8ad2f6..c3dde8ef1 100644
--- a/src/settings.js
+++ b/src/settings.js
@@ -2206,7 +2206,7 @@ var CROSS_ORIGIN = false;
 // produces a normal object file (i.e. ``ld -r``).  When set to true (the
 // default) the ``-shared`` flag is equivelent to :ref:`SIDE_MODULE` and will
 // produce a Wasn dynamic library.
-var FAKE_DYLIBS = false;
+var FAKE_DYLIBS = true;
 
 // Add a #! line to generated JS file and make it executable.  This is useful
 // for building command line tools that run under node.
-- 
2.54.0

