From b2186eb1985c1ae793393ce43139ae85f967113b Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Mon, 3 Feb 2025 17:10:11 +0100
Subject: [PATCH 6/6] Verbose

---
 em++ | 2 ++
 emcc | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/em++ b/em++
index 0ca8082a2..d684a13a4 100755
--- a/em++
+++ b/em++
@@ -32,6 +32,8 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
+echo `basename $0` $@
+
 if [ -z "$_EMCC_CCACHE" ]; then
   exec "$PYTHON" -E "$0.py" "$@"
 else
diff --git a/emcc b/emcc
index 0ca8082a2..d684a13a4 100755
--- a/emcc
+++ b/emcc
@@ -32,6 +32,8 @@ if [ -z "$PYTHON" ]; then
   exit 1
 fi
 
+echo `basename $0` $@
+
 if [ -z "$_EMCC_CCACHE" ]; then
   exec "$PYTHON" -E "$0.py" "$@"
 else
-- 
2.34.1

