From 9e76b6d46968d028c1c8370fcdc8e7be84da47b6 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Wed, 29 Jan 2025 11:05:27 +0100
Subject: [PATCH 6/6] Record the exitStatus when exit is called

---
 src/shell.js | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/shell.js b/src/shell.js
index d2334a83d..7036e79f2 100644
--- a/src/shell.js
+++ b/src/shell.js
@@ -171,6 +171,7 @@ var moduleOverrides = Object.assign({}, Module);
 var arguments_ = [];
 var thisProgram = './this.program';
 var quit_ = (status, toThrow) => {
+  Module.exitCode = status;
   throw toThrow;
 };
 
@@ -281,6 +282,7 @@ if (ENVIRONMENT_IS_NODE) {
 #endif
 
   quit_ = (status, toThrow) => {
+    Module.exitCode = status;
     process.exitCode = status;
     throw toThrow;
   };
-- 
2.34.1

