From b234f241e1a9937d4697b6a421c9d154197b6e55 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Fri, 31 Jan 2025 11:42:59 +0100
Subject: [PATCH 6/6] sdl2_image wasm-sjlj variant

---
 tools/ports/sdl2_image.py | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/tools/ports/sdl2_image.py b/tools/ports/sdl2_image.py
index c72ef576b..0d3b758bd 100644
--- a/tools/ports/sdl2_image.py
+++ b/tools/ports/sdl2_image.py
@@ -42,6 +42,8 @@ def get_lib_name(settings):
   libname = 'libSDL2_image'
   if formats != '':
     libname += '_' + formats
+  if settings.SUPPORT_LONGJMP == 'wasm':
+    libname += '-wasm-sjlj'
   return libname + '.a'
 
 
@@ -58,20 +60,23 @@ def get(ports, settings, shared):
               IMG_tif.c IMG_xcf.c IMG_xpm.c IMG_xv.c IMG_webp.c IMG_ImageIO.m
               IMG_avif.c IMG_jxl.c IMG_svg.c IMG_qoi.c'''.split()
 
-    defs = ['-O2', '-sUSE_SDL=2', '-Wno-format-security']
+    flags = ['-O2', '-sUSE_SDL=2', '-Wno-format-security']
 
     formats = get_formats(settings)
 
     for fmt in formats:
-      defs.append('-DLOAD_' + fmt.upper())
+      flags.append('-DLOAD_' + fmt.upper())
 
     if 'png' in formats:
-      defs += ['-sUSE_LIBPNG']
+      flags += ['-sUSE_LIBPNG']
 
     if 'jpg' in formats:
-      defs += ['-sUSE_LIBJPEG']
+      flags += ['-sUSE_LIBJPEG']
 
-    ports.build_port(src_dir, final, 'sdl2_image', flags=defs, srcs=srcs)
+    if settings.SUPPORT_LONGJMP == 'wasm':
+      flags.append('-sSUPPORT_LONGJMP=wasm')
+
+    ports.build_port(src_dir, final, 'sdl2_image', flags=flags, srcs=srcs)
 
   return [shared.cache.get_lib(libname, create, what='port')]
 
-- 
2.34.1

