--- emsdk/emsdk/emscripten/tag-1.38.31/emcc.py	2019-04-24 16:21:45.000000000 +0200
+++ emsdk/emsdk/emscripten/tag-1.38.31/emcc.py	2019-04-24 16:21:45.000000000 +0200
@@ -816,7 +816,7 @@
           newargs[i] = newargs[i + 1] = ''
           if key == 'WASM_BACKEND=1':
             exit_with_error('do not set -s WASM_BACKEND, instead set EMCC_WASM_BACKEND=1 in the environment')
-    newargs = [arg for arg in newargs if arg is not '']
+    newargs = [arg for arg in newargs if arg != '']
 
     settings_key_changes = set()
     for s in settings_changes:
@@ -936,7 +936,7 @@
 
     original_input_files = input_files[:]
 
-    newargs = [a for a in newargs if a is not '']
+    newargs = [a for a in newargs if a != '']
 
     # -c means do not link in gcc, and for us, the parallel is to not go all the way to JS, but stop at bitcode
     has_dash_c = '-c' in newargs
