--- a/emsdk/upstream/emscripten/src/library_syscall.js
+++ b/emsdk/upstream/emscripten/src/library_syscall.js
@@ -755,7 +755,7 @@ var SyscallsLibrary = {
   },
   __sys_wait4__proxy: false,
   __sys_wait4: function(pid, wstart, options, rusage) {
-    abort('cannot wait on child processes');
+    return -({{{ cDefine('ENOSYS') }}});
   },
   __sys_setdomainname__nothrow: true,
   __sys_setdomainname__proxy: false,
