--- a/emsdk/binaryen/tag-1.38.31/src/passes/FuncCastEmulation.cpp	2019-04-03 11:30:13.556074729 -0400
+++ b/emsdk/binaryen/tag-1.38.31/src/passes/FuncCastEmulation.cpp	2019-04-03 11:30:32.817143862 -0400
@@ -39,7 +39,7 @@
 // This should be enough for everybody. (As described above, we need this
 // to match when dynamically linking, and also dynamic linking is why we
 // can't just detect this automatically in the module we see.)
-static const int NUM_PARAMS = 16;
+static const int NUM_PARAMS = 61;   // needed by scipy.odr._odrpack
 
 // Converts a value to the ABI type of i64.
 static Expression* toABI(Expression* value, Module* module) {
