From 9ed9022d88cb05e46b3d1aa077ba9e4d4ec740f8 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Tue, 30 Aug 2022 10:58:25 -0700
Subject: [PATCH 08/10] Remove meson requirements

---
 pyproject.toml | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index ccab62aec..6f201cac8 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -8,9 +8,8 @@
 #     "pybind11>=2.4.3,<2.5.0",
 
 [build-system]
-build-backend = 'mesonpy'
 requires = [
-    "meson-python>=0.12.1,<0.14.0", # already working with 0.13.x series at branch time
+    "setuptools<60.0.0",
     "Cython>=0.29.35,<3.0",  # when updating version, also update check in meson.build
     "pybind11>=2.10.4,<2.11.0",
     "pythran>=0.12.0,<0.14.0", # already working with 0.13.x series at branch time
-- 
2.41.0

