From 4d4479b0af5041d2c4fcf0158ffa51caa7658bae Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Est=C3=A8ve?= <loic.esteve@ymail.com>
Date: Thu, 6 Apr 2023 12:22:45 +0200
Subject: [PATCH 11/12] Enable long double support in Boost library for
 emscripten

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index ff602a1cd..1a408751f 100755
--- a/setup.py
+++ b/setup.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python 
 """SciPy: Scientific Library for Python
 
 SciPy (pronounced "Sigh Pie") is open-source software for mathematics,
-- 
2.25.1

