From fe748bc50293e7bd938d6dba8896e3adb6b6b5e0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Est=C3=A8ve?= <loic.esteve@ymail.com>
Date: Sat, 4 Feb 2023 06:15:31 +0100
Subject: [PATCH] Build only selected packages

---
 scipy/setup.py | 30 +++++++++++++++---------------
 1 file changed, 15 insertions(+), 15 deletions(-)

diff --git a/scipy/setup.py b/scipy/setup.py
index 39a03cdb7..85840aa62 100644
--- a/scipy/setup.py
+++ b/scipy/setup.py
@@ -4,24 +4,24 @@ def configuration(parent_package='',top_path=None):
 
     from numpy.distutils.misc_util import Configuration
     config = Configuration('scipy',parent_package,top_path)
-    config.add_subpackage('cluster')
-    config.add_subpackage('constants')
-    config.add_subpackage('fft')
-    config.add_subpackage('fftpack')
-    config.add_subpackage('integrate')
-    config.add_subpackage('interpolate')
-    config.add_subpackage('io')
+    # config.add_subpackage('cluster')
+    # config.add_subpackage('constants')
+    # config.add_subpackage('fft')
+    # config.add_subpackage('fftpack')
+    # config.add_subpackage('integrate')
+    # config.add_subpackage('interpolate')
+    # config.add_subpackage('io')
     config.add_subpackage('linalg')
     config.add_data_files('*.pxd')
-    config.add_subpackage('misc')
-    config.add_subpackage('odr')
-    config.add_subpackage('optimize')
-    config.add_subpackage('signal')
+    # config.add_subpackage('misc')
+    # config.add_subpackage('odr')
+    # config.add_subpackage('optimize')
+    # config.add_subpackage('signal')
     config.add_subpackage('sparse')
-    config.add_subpackage('spatial')
-    config.add_subpackage('special')
-    config.add_subpackage('stats')
-    config.add_subpackage('ndimage')
+    # config.add_subpackage('spatial')
+    # config.add_subpackage('special')
+    # config.add_subpackage('stats')
+    # config.add_subpackage('ndimage')
     config.add_subpackage('_build_utils')
     config.add_subpackage('_lib')
     config.make_config_py()
-- 
2.34.1

