commit 15e9f01458eec8cf9fc43de6646ee75cfa361859
Author: Roman Yurchak <rth.yurchak@pm.me>
Date:   Wed Oct 10 14:12:22 2018 +0200

    Remove modules with global C++ namespace issue

diff --git a/scipy/interpolate/setup.py b/scipy/interpolate/setup.py
index b42d3345c..2c8279403 100755
--- a/scipy/interpolate/setup.py
+++ b/scipy/interpolate/setup.py
@@ -35,10 +35,10 @@ def configuration(parent_package='',top_path=None):
                          depends=fitpack_src,
                          )
 
-    config.add_extension('_interpolate',
-                         sources=['src/_interpolate.cpp'],
-                         include_dirs=['src'],
-                         depends=['src/interpolate.h'])
+    #config.add_extension('_interpolate',
+    #                     sources=['src/_interpolate.cpp'],
+    #                     include_dirs=['src'],
+    #                     depends=['src/interpolate.h'])
 
     config.add_data_dir('tests')
 
diff --git a/scipy/sparse/setup.py b/scipy/sparse/setup.py
index ecf8ae5c6..27fd128c3 100755
--- a/scipy/sparse/setup.py
+++ b/scipy/sparse/setup.py
@@ -45,17 +45,17 @@ def configuration(parent_package='',top_path=None):
                'sparsetools.h',
                'util.h']
     depends = [os.path.join('sparsetools', hdr) for hdr in depends],
-    config.add_extension('_sparsetools',
-                         define_macros=[('__STDC_FORMAT_MACROS', 1)],
-                         depends=depends,
-                         include_dirs=['sparsetools'],
-                         sources=[os.path.join('sparsetools', 'sparsetools.cxx'),
-                                  os.path.join('sparsetools', 'csr.cxx'),
-                                  os.path.join('sparsetools', 'csc.cxx'),
-                                  os.path.join('sparsetools', 'bsr.cxx'),
-                                  os.path.join('sparsetools', 'other.cxx'),
-                                  get_sparsetools_sources]
-                         )
+    #config.add_extension('_sparsetools',
+    #                     define_macros=[('__STDC_FORMAT_MACROS', 1)],
+    #                     depends=depends,
+    #                     include_dirs=['sparsetools'],
+    #                     sources=[os.path.join('sparsetools', 'sparsetools.cxx'),
+    #                              os.path.join('sparsetools', 'csr.cxx'),
+    #                              os.path.join('sparsetools', 'csc.cxx'),
+    #                              os.path.join('sparsetools', 'bsr.cxx'),
+    #                              os.path.join('sparsetools', 'other.cxx'),
+    #                              get_sparsetools_sources]
+    #                     )
 
     return config
 
diff --git a/scipy/spatial/setup.py b/scipy/spatial/setup.py
index 43b635dec..5b100ac06 100755
--- a/scipy/spatial/setup.py
+++ b/scipy/spatial/setup.py
@@ -58,10 +58,10 @@ def configuration(parent_package='', top_path=None):
     ckdtree_headers = [join('ckdtree', 'src', x) for x in ckdtree_headers]
         
     ckdtree_dep = ['ckdtree.cxx'] + ckdtree_headers + ckdtree_src
-    config.add_extension('ckdtree',
-                         sources=['ckdtree.cxx'] + ckdtree_src,
-                         depends=ckdtree_dep,
-                         include_dirs=inc_dirs + [join('ckdtree','src')])
+    #config.add_extension('ckdtree',
+    #                     sources=['ckdtree.cxx'] + ckdtree_src,
+    #                     depends=ckdtree_dep,
+    #                     include_dirs=inc_dirs + [join('ckdtree','src')])
     # _distance_wrap
     config.add_extension('_distance_wrap',
         sources=[join('src', 'distance_wrap.c')],
diff --git a/scipy/special/setup.py b/scipy/special/setup.py
index e5efb2e7a..6f75681c0 100755
--- a/scipy/special/setup.py
+++ b/scipy/special/setup.py
@@ -91,18 +91,18 @@ def configuration(parent_package='',top_path=None):
                       '_faddeeva.cxx', 'Faddeeva.cc']
     ufuncs_cxx_dep = (headers + ufuncs_cxx_src + cephes_src
                       + ['*.hh'])
-    config.add_extension('_ufuncs_cxx',
-                         sources=ufuncs_cxx_src,
-                         depends=ufuncs_cxx_dep,
-                         include_dirs=[curdir],
-                         define_macros=define_macros,
-                         extra_info=get_info("npymath"))
-
-    cfg = dict(get_system_info('lapack_opt'))
-    config.add_extension('_ellip_harm_2',
-                         sources=['_ellip_harm_2.c', 'sf_error.c',],
-                         **cfg
-                         )
+    #config.add_extension('_ufuncs_cxx',
+    #                     sources=ufuncs_cxx_src,
+    #                     depends=ufuncs_cxx_dep,
+    #                     include_dirs=[curdir],
+    #                     define_macros=define_macros,
+    #                     extra_info=get_info("npymath"))
+
+    #cfg = dict(get_system_info('lapack_opt'))
+    #config.add_extension('_ellip_harm_2',
+    #                     sources=['_ellip_harm_2.c', 'sf_error.c',],
+    #                     **cfg
+    #                     )
 
     config.add_data_files('tests/*.py')
     config.add_data_files('tests/data/README')
