commit 7352fc4a977a3414cf3194faf6ef990638a6329c
Author: Roman Yurchak <rth.yurchak@gmail.com>
Date:   Sat Feb 27 11:13:29 2021 +0100

    fix-blas.patch

diff --git a/setup.py b/setup.py
index 0e11de60f..8c28c613d 100755
--- a/setup.py
+++ b/setup.py
@@ -484,7 +484,7 @@ def configuration(parent_package='', top_path=None):
 
     lapack_opt = get_info('lapack_opt')
 
-    if not lapack_opt:
+    if not lapack_opt and False:
         if sys.platform == "darwin":
             msg = ('No BLAS/LAPACK libraries found. '
                    'Note: Accelerate is no longer supported.')
