diff --git a/../../../intdy.c b/intdy.c
index 78c069c..003177e 100644
--- a/../../../intdy.c
+++ b/intdy.c
@@ -51,7 +51,7 @@ static integer c__2 = 2;
     static integer jb2, jj1, jp1;
     extern /* Subroutine */ int xerrwv_(char *, integer *, integer *, integer 
 	    *, integer *, integer *, integer *, integer *, doublereal *, 
-	    doublereal *, ftnlen);
+	    doublereal *);
 
 /* lll. optimize */
 /* ----------------------------------------------------------------------- */
@@ -151,15 +151,15 @@ L55:
 
 L80:
     xerrwv_("intdy--  k (=i1) illegal      ", &c__30, &c__51, &c__0, &c__1, k,
-	     &c__0, &c__0, &c_b20, &c_b20, (ftnlen)30);
+	     &c__0, &c__0, &c_b20, &c_b20);
     *iflag = -1;
     return 0;
 L90:
     xerrwv_("intdy--  t (=r1) illegal      ", &c__30, &c__52, &c__0, &c__0, &
-	    c__0, &c__0, &c__1, t, &c_b20, (ftnlen)30);
+	    c__0, &c__0, &c__1, t, &c_b20);
     xerrwv_("     t not in interval tcur - hu (= r1) to tcur (=r2)       ", &
 	    c__60, &c__52, &c__0, &c__0, &c__0, &c__0, &c__2, &tp, &
-	    ls0001_1.tn, (ftnlen)60);
+	    ls0001_1.tn);
     *iflag = -2;
     return 0;
 /* ----------------------- end of subroutine intdy ----------------------- */
