From 7b56e1341c486422973dfa890158872200527b51 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Sat, 25 Dec 2021 20:31:40 -0800
Subject: [PATCH] Remove cuncsd, dorcsd, sorcsd, zuncsd

---
 scipy/linalg/__init__.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/scipy/linalg/__init__.py b/scipy/linalg/__init__.py
index e88479b48..746bf3dce 100644
--- a/scipy/linalg/__init__.py
+++ b/scipy/linalg/__init__.py
@@ -191,6 +191,12 @@ Low-level routines
    `scipy.linalg.cython_lapack` -- Low-level LAPACK functions for Cython
 
 """  # noqa: E501
+from . import lapack
+del lapack.cuncsd
+del lapack.dorcsd
+del lapack.sorcsd
+del lapack.zuncsd
+
 
 from .misc import *
 from .basic import *
-- 
2.25.1

