commit 98f6cef78f08a992c5dcf746bd2802ad0e570df4
Author: Roman Yurchak <rth.yurchak@pm.me>
Date:   Sat Oct 31 22:41:43 2020 +0100

    Skip _ellip_harm_2.py due to ctypes

diff --git a/scipy/special/__init__.py b/scipy/special/__init__.py
index a152e17dd..303298ec1 100644
--- a/scipy/special/__init__.py
+++ b/scipy/special/__init__.py
@@ -645,7 +645,7 @@ from . import specfun
 from . import orthogonal
 from .orthogonal import *
 from .spfun_stats import multigammaln
-from ._ellip_harm import ellip_harm, ellip_harm_2, ellip_normal
+from ._ellip_harm import ellip_harm, ellip_normal
 from .lambertw import lambertw
 from ._spherical_bessel import (spherical_jn, spherical_yn, spherical_in,
                                 spherical_kn)
diff --git a/scipy/special/_ellip_harm.py b/scipy/special/_ellip_harm.py
index 23af1124e..a94e56489 100644
--- a/scipy/special/_ellip_harm.py
+++ b/scipy/special/_ellip_harm.py
@@ -4,7 +4,15 @@ import threading
 import numpy as np
 
 from ._ufuncs import _ellip_harm
-from ._ellip_harm_2 import _ellipsoid, _ellipsoid_norm
+# from ._ellip_harm_2 import _ellipsoid, _ellipsoid_norm
+
+
+def _raise_error(*cargs, **kwargs):
+    raise NotImplementedError
+
+
+_ellipsoid = _raise_error
+_ellipsoid_norm = _raise_error
 
 
 def ellip_harm(h2, k2, n, p, s, signm=1, signn=1):
diff --git a/scipy/special/setup.py b/scipy/special/setup.py
index 12b988f3f..55e33ffbc 100644
--- a/scipy/special/setup.py
+++ b/scipy/special/setup.py
@@ -97,11 +97,11 @@ def configuration(parent_package='',top_path=None):
                          define_macros=define_macros,
                          extra_info=get_info("npymath"))
 
-    cfg = dict(get_system_info('lapack_opt'))
-    config.add_extension('_ellip_harm_2',
-                         sources=['_ellip_harm_2.c', 'sf_error.c',],
-                         **cfg
-                         )
+    # cfg = dict(get_system_info('lapack_opt'))
+    # config.add_extension('_ellip_harm_2',
+    #                      sources=['_ellip_harm_2.c', 'sf_error.c',],
+    #                      **cfg
+    #                      )
 
     # Cython API
     config.add_data_files('cython_special.pxd')
