commit 066faced67d1d0a8a4a89588194fcbd2faa0a525
Author: Roman Yurchak <rth.yurchak@pm.me>
Date:   Wed Oct 10 17:48:17 2018 +0200

    Skip scipy.special._ufuncs

diff --git a/scipy/special/__init__.py b/scipy/special/__init__.py
index f013b818e..ab4956825 100644
--- a/scipy/special/__init__.py
+++ b/scipy/special/__init__.py
@@ -624,7 +624,11 @@ Convenience Functions
 
 from __future__ import division, print_function, absolute_import
 
-from ._ufuncs import *
+try:
+    from ._ufuncs import *
+except ImportError:
+    # TODO
+    pass
 
 from .basic import *
 from . import specfun
diff --git a/scipy/special/_ellip_harm.py b/scipy/special/_ellip_harm.py
index 6972c2ad1..0558ee6d3 100644
--- a/scipy/special/_ellip_harm.py
+++ b/scipy/special/_ellip_harm.py
@@ -3,7 +3,11 @@ from __future__ import division, print_function, absolute_import
 import threading
 import numpy as np
 
-from ._ufuncs import _ellip_harm
+try:
+    from ._ufuncs import _ellip_harm
+except ImportError:
+    # TODO
+    pass
 from ._ellip_harm_2 import _ellipsoid, _ellipsoid_norm
 
 
diff --git a/scipy/special/basic.py b/scipy/special/basic.py
index bb251e745..f1c788c3d 100644
--- a/scipy/special/basic.py
+++ b/scipy/special/basic.py
@@ -11,9 +11,13 @@ from scipy._lib.six import xrange
 from numpy import (pi, asarray, floor, isscalar, iscomplex, real, imag, sqrt,
                    where, mgrid, sin, place, issubdtype, extract,
                    less, inexact, nan, zeros, atleast_1d, sinc)
-from ._ufuncs import (ellipkm1, mathieu_a, mathieu_b, iv, jv, gamma, psi, zeta,
-                      hankel1, hankel2, yv, kv, gammaln, ndtri, errprint, poch,
-                      binom)
+try:
+    from ._ufuncs import (ellipkm1, mathieu_a, mathieu_b, iv, jv, gamma, psi, zeta,
+                          hankel1, hankel2, yv, kv, gammaln, ndtri, errprint, poch,
+                          binom)
+except ImportError:
+    # TODO
+    pass
 from . import specfun
 from . import orthogonal
 
diff --git a/scipy/special/lambertw.py b/scipy/special/lambertw.py
index 4bdac30f9..8a890ce51 100644
--- a/scipy/special/lambertw.py
+++ b/scipy/special/lambertw.py
@@ -1,6 +1,10 @@
 from __future__ import division, print_function, absolute_import
 
-from ._ufuncs import _lambertw
+try:
+    from ._ufuncs import _lambertw
+except ImportError:
+    # TODO
+    pass
 
 
 def lambertw(z, k=0, tol=1e-8):
diff --git a/scipy/special/orthogonal.py b/scipy/special/orthogonal.py
index c2b8dead3..b33ab96ac 100755
--- a/scipy/special/orthogonal.py
+++ b/scipy/special/orthogonal.py
@@ -102,8 +102,12 @@ from scipy import linalg
 from scipy.special import airy
 
 # Local imports.
-from . import _ufuncs as cephes
-_gam = cephes.gamma
+try:
+    from . import _ufuncs as cephes
+    _gam = cephes.gamma
+except ImportError:
+    # TODO
+    pass
 from . import specfun
 
 __all__ = ['legendre', 'chebyt', 'chebyu', 'chebyc', 'chebys',
@@ -1657,8 +1661,12 @@ def sh_legendre(n, monic=False):
 # -----------------------------------------------------------------------------
 # Vectorized functions for evaluation
 # -----------------------------------------------------------------------------
-from ._ufuncs import (binom, eval_jacobi, eval_sh_jacobi, eval_gegenbauer,
-                      eval_chebyt, eval_chebyu, eval_chebys, eval_chebyc,
-                      eval_sh_chebyt, eval_sh_chebyu, eval_legendre,
-                      eval_sh_legendre, eval_genlaguerre, eval_laguerre,
-                      eval_hermite, eval_hermitenorm)
+try:
+    from ._ufuncs import (binom, eval_jacobi, eval_sh_jacobi, eval_gegenbauer,
+                          eval_chebyt, eval_chebyu, eval_chebys, eval_chebyc,
+                          eval_sh_chebyt, eval_sh_chebyu, eval_legendre,
+                          eval_sh_legendre, eval_genlaguerre, eval_laguerre,
+                          eval_hermite, eval_hermitenorm)
+except ImportError:
+    # TODO
+    pass
