From 53ee3cee6aae5333d61ac687677cf2ca39cb831a Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Fri, 8 Apr 2022 18:17:51 -0700
Subject: [PATCH 1/2] Use patched chrono

---
 src/rust/Cargo.toml | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/rust/Cargo.toml b/src/rust/Cargo.toml
index 617167d0..35bf38ae 100644
--- a/src/rust/Cargo.toml
+++ b/src/rust/Cargo.toml
@@ -5,6 +5,9 @@ authors = ["The cryptography developers <cryptography-dev@python.org>"]
 edition = "2018"
 publish = false
 
+[patch.crates-io]
+chrono = { path =  "../../chrono" }
+
 [dependencies]
 lazy_static = "1"
 pyo3 = { version = "0.15.1" }
-- 
2.25.1

