From 740913f58345581d1739eb976ce73394356ad683 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Sun, 28 Aug 2022 10:00:45 -0700
Subject: [PATCH] Fix compilation with -Wincompatible-function-pointer-types

---
 erfa/ufunc.c.templ | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/erfa/ufunc.c.templ b/erfa/ufunc.c.templ
index a24239a..934ca79 100644
--- a/erfa/ufunc.c.templ
+++ b/erfa/ufunc.c.templ
@@ -142,7 +142,7 @@ static inline void copy_to_eraLDBODY(char *ptr, npy_intp s, npy_intp n,
 {%- for func in funcs %}
 
 static void ufunc_loop_{{ func.pyname }}(
-    char **args, npy_intp *dimensions, npy_intp* steps, void* data)
+    char **args, const npy_intp *dimensions, const npy_intp* steps, void* data)
 {
   {#- /* index and length of loop */ #}
     npy_intp i_o;
-- 
2.25.1

