From befdbb64fc403ff383ea656f9b237f3a989e8603 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Wed, 15 Feb 2023 14:24:37 +0100
Subject: [PATCH 1/5] Make Configure script verbose

---
 Configure | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Configure b/Configure
index 0cbbc6b..d0516ae 100755
--- a/Configure
+++ b/Configure
@@ -15,6 +15,7 @@
 #
 # Perl's Configure and GNU autoconfig were of much help in writing these files.
 # $Id$
+set -v

 TOP=`pwd`
 MAKE=${MAKE:-make}
--
2.39.1
