diff --git a/protobuf-5.29.0/google/protobuf/service.py b/protobuf-5.29.0/google/protobuf/service.py
index 8002c04..795b602 100644
--- a/protobuf-5.29.0/google/protobuf/service.py
+++ b/protobuf-5.29.0/google/protobuf/service.py
@@ -75,7 +75,7 @@ class Service(object):
      Further details about the failure can be found by querying the
      RpcController.
     """
-   raise NotImplementedError
+    raise NotImplementedError
 
   def GetRequestClass(self, method_descriptor):
     """Returns the class of the request message for the specified method.
