diff --git a/python/message.c b/python/message.c
index 66c26f4..994d714 100644
--- a/python/message.c
+++ b/python/message.c
@@ -1235,7 +1235,7 @@ static PyObject* PyUpb_Message_CopyFrom(PyObject* _self, PyObject* arg) {
   Py_DECREF(tmp);
 
   upb_Message_DeepCopy(self->ptr.msg, other->ptr.msg,
-                       upb_MessageDef_MiniTable(other->def),
+                       upb_MessageDef_MiniTable(&other->def),
                        PyUpb_Arena_Get(self->arena));
   PyUpb_Message_SyncSubobjs(self);
 
@@ -1608,9 +1608,9 @@ PyObject* DeepCopy(PyObject* _self, PyObject* arg) {
 
   PyObject* arena = PyUpb_Arena_New();
   upb_Message* clone =
-      upb_Message_DeepClone(self->ptr.msg, upb_MessageDef_MiniTable(self->def),
+      upb_Message_DeepClone(self->ptr.msg, upb_MessageDef_MiniTable(&self->def),
                             PyUpb_Arena_Get(arena));
-  PyObject* ret = PyUpb_Message_Get(clone, self->def, arena);
+  PyObject* ret = PyUpb_Message_Get(clone, &self->def, arena);
   Py_DECREF(arena);
 
   return ret;
