From e19976d9489eb01fe6c7347bb09b118101aa1fcc Mon Sep 17 00:00:00 2001
From: Julius Michaelis <gitter@liftm.de>
Date: Sun, 19 Apr 2020 12:23:58 +0900
Subject: [PATCH] em-replay-library-path

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 4b48767..acfbd09 100644
--- a/setup.py
+++ b/setup.py
@@ -66,7 +66,7 @@ def maybe_build_libmecab_and_adjust_flags(ext):
             SRCDIR
         ])
         inc_dir  = [os.path.join(SRCDIR, "build/libmecab/mecab/src")]
-        lib_dirs = [os.path.join(SRCDIR, "build/libmecab/mecab/src")]
+        lib_dirs = [os.path.join(SRCDIR, "build/libmecab/mecab/src/.libs")]
 
         # mecab-config --libs-only-l will produce the list of
         # libraries needed to link with a hypothetical *shared*
-- 
2.26.0

