From f5d4c338ffa7ef8f1a6f70be9bde973164bea849 Mon Sep 17 00:00:00 2001
From: Gyeongjae Choi <def6488@gmail.com>
Date: Tue, 16 Aug 2022 14:17:31 +0900
Subject: [PATCH] CMake: fix typo in fuzzer CMakeLists.txt

---
 fuzzers/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fuzzers/CMakeLists.txt b/fuzzers/CMakeLists.txt
index b1a1825653..517db4a01e 100644
--- a/fuzzers/CMakeLists.txt
+++ b/fuzzers/CMakeLists.txt
@@ -3,7 +3,7 @@
 if (NOT TARGET ogr_SQLite)
   return()
 endif ()
-if (CMAKE_CROSSCOMPILONG OR WIN32)
+if (CMAKE_CROSSCOMPILING OR WIN32)
   return()
 endif ()
 if (NOT CMAKE_VERSION VERSION_GREATER_EQUAL 3.12)
-- 
2.37.0

