From 7be4072905afbf3f5571d8fcac1a3e8402edf0a5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Est=C3=A8ve?= <loic.esteve@ymail.com>
Date: Mon, 22 May 2023 11:30:06 +0200
Subject: [PATCH] Add meson debug

---
 pyproject.toml | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/pyproject.toml b/pyproject.toml
index cdc557a00..c00899490 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -229,3 +229,6 @@ python_functions = [
 ]
 [tool.coverage.run]
 omit = ['*/tests/*']
+
+[tool.meson-python.args]
+compile=['-v']
-- 
2.34.1

