From a116131b3605cfdd0bea2816c5381df2cb87a686 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Est=C3=A8ve?= <loic.esteve@ymail.com>
Date: Mon, 22 May 2023 11:30:06 +0200
Subject: [PATCH] Add meson debug

---
 pyproject.toml | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/pyproject.toml b/pyproject.toml
index cdc557a00..542f85a88 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -229,3 +229,7 @@ python_functions = [
 ]
 [tool.coverage.run]
 omit = ['*/tests/*']
+
+[tool.meson-python.args]
+compile=['-v']
+setup=['--cross-file', '$PYODIDE_ROOT/tools/emscripten.meson.cross']
-- 
2.34.1

