From 7e50329062af2b51d0e551c6c80fc5b457263639 Mon Sep 17 00:00:00 2001
From: Hood <hood@mit.edu>
Date: Tue, 15 Jun 2021 09:06:01 -0700
Subject: [PATCH] Remove ctypes import

---
 setuptools/windows_support.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setuptools/windows_support.py b/setuptools/windows_support.py
index cb977cf..0324739 100644
--- a/setuptools/windows_support.py
+++ b/setuptools/windows_support.py
@@ -1,5 +1,4 @@
 import platform
-import ctypes
 
 
 def windows_only(func):
-- 
2.17.1

