diff --git a/setup.py b/setup.py
index d8c1872a..ad396e02 100644
--- a/setup.py
+++ b/setup.py
@@ -167,7 +167,6 @@ def get_extension_modules():
     ext_options = {
         "include_dirs": include_dirs,
         "library_dirs": library_dirs,
-        "runtime_library_dirs": library_dirs if os.name != "nt" else None,
         "libraries": get_libraries(library_dirs),
     }
     # setup cythonized modules
