From 892d1e37b1c1779c0d14f86b507971c26cc18a2d Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Fri, 10 Dec 2021 07:37:17 -0800
Subject: [PATCH] Respect skip build flag

---
 setup.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index fbf76a1..fa9e7a6 100644
--- a/setup.py
+++ b/setup.py
@@ -70,7 +70,8 @@ try:
 
     class CustomBdistWheel(bdist_wheel):
         def run(self):
-            self.run_command('build_ext')
+            if not self.skip_build:
+                self.run_command('build_ext')
             bdist_wheel.run(self)
 
     custom_cmd_class['bdist_wheel'] = CustomBdistWheel
-- 
2.25.1

