From 4e4f12d0052cb9c9fc8ac550266889a0c39af8ec Mon Sep 17 00:00:00 2001
From: Jo Bovy <bovy@astro.utoronto.ca>
Date: Mon, 28 Feb 2022 20:43:45 -0500
Subject: [PATCH] Move ssl import into functions to avoid import error

---
 astropy/utils/data.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/astropy/utils/data.py b/astropy/utils/data.py
index 0dde5b293..25c758be2 100644
--- a/astropy/utils/data.py
+++ b/astropy/utils/data.py
@@ -13,7 +13,6 @@ import io
 import re
 import shutil
 import socket
-import ssl
 import sys
 import urllib.request
 import urllib.error
@@ -1048,6 +1047,7 @@ def _build_urlopener(ftp_tls=False, ssl_context=None, allow_insecure=False):
     """
     Helper for building a `urllib.request.build_opener` which handles TLS/SSL.
     """
+    import ssl
 
     ssl_context = dict(it for it in ssl_context) if ssl_context else {}
     cert_chain = {}
@@ -1087,6 +1087,7 @@ def _build_urlopener(ftp_tls=False, ssl_context=None, allow_insecure=False):
 def _try_url_open(source_url, timeout=None, http_headers=None, ftp_tls=False,
                   ssl_context=None, allow_insecure=False):
     """Helper for opening a URL while handling TLS/SSL verification issues."""
+    import ssl
 
     # Always try first with a secure connection
     # _build_urlopener uses lru_cache, so the ssl_context argument must be
-- 
2.32.0 (Apple Git-132)

