diff --git a/sklearn/_build_utils/__init__.py b/sklearn/_build_utils/__init__.py
index 8c57982ca..b8eb93545 100644
--- a/sklearn/_build_utils/__init__.py
+++ b/sklearn/_build_utils/__init__.py
@@ -47,7 +47,7 @@ def cythonize_extensions(top_path, config):
 
     # Fast fail before cythonization if compiler fails compiling basic test
     # code even without OpenMP
-    basic_check_build()
+    # basic_check_build()
 
     # check simple compilation with OpenMP. If it fails scikit-learn will be
     # built without OpenMP and the test test_openmp_supported in the test suite
@@ -61,7 +61,7 @@ def cythonize_extensions(top_path, config):
     #   cython build-time variable passed to the cythonize() call.
     # - Then in the build_ext subclass defined in the top-level setup.py file
     #   to actually build the compiled extensions with OpenMP flags if needed.
-    sklearn._OPENMP_SUPPORTED = check_openmp_support()
+    sklearn._OPENMP_SUPPORTED = False  # check_openmp_support()
 
     n_jobs = 1
     with contextlib.suppress(ImportError):
