commit 55cfffe8243d7d85dc6f8d14f045cee8a4e8b0ae
Author: Roman Yurchak <rth.yurchak@pm.me>
Date:   Mon Dec 3 22:09:46 2018 +0100

    Use site joblib

diff --git a/sklearn/utils/_joblib.py b/sklearn/utils/_joblib.py
index c3df15e91..b9f218ef1 100644
--- a/sklearn/utils/_joblib.py
+++ b/sklearn/utils/_joblib.py
@@ -5,7 +5,7 @@ import os as _os
 import warnings as _warnings
 
 # An environment variable to use the site joblib
-if _os.environ.get('SKLEARN_SITE_JOBLIB', False):
+if True:
     with _warnings.catch_warnings():
         _warnings.simplefilter("ignore")
         # joblib imports may raise DeprecationWarning on certain Python
