diff --git a/imageio/core/findlib.py b/imageio/core/findlib.py
--- a/imageio/core/findlib.py
+++ b/imageio/core/findlib.py
@@ -7,7 +7,6 @@
 
 import os
 import sys
-import ctypes
 
 
 LOCALDIR = os.path.abspath(os.path.dirname(__file__))
@@ -95,6 +94,7 @@ def load_lib(exact_lib_names, lib_names, lib_dirs=None):
 
     Returns ``(ctypes_library, library_path)``
     """
+    import ctypes
 
     # Checks
     assert isinstance(exact_lib_names, list)

diff --git a/imageio/plugins/__init__.py b/imageio/plugins/__init__.py
index 7e26683..971f83e 100644
--- a/imageio/plugins/__init__.py
+++ b/imageio/plugins/__init__.py
@@ -84,8 +84,8 @@ from . import tifffile
 from . import pillow
 from . import grab
 
-from . import freeimage
-from . import freeimagemulti
+# from . import freeimage
+# from . import freeimagemulti
 
 from . import ffmpeg
 
