From 1470dc5c4611d383510d6184ea02e893b5723349 Mon Sep 17 00:00:00 2001
From: Ubuntu <ubuntu@ip-172-31-56-197.us-west-2.compute.internal>
Date: Wed, 7 Sep 2022 16:21:33 +0000
Subject: [PATCH] noomp=True

---
 setup.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/setup.py b/setup.py
index 56c67ff..0812c03 100644
--- a/setup.py
+++ b/setup.py
@@ -112,7 +112,7 @@ def compile_cpp(
     boost_librarydir: Optional[str] = None,
     opencl_include_dir: Optional[str] = None,
     opencl_library: Optional[str] = None,
-    nomp: bool = False,
+    nomp: bool = True,
     bit32: bool = False,
     integrated_opencl: bool = False
 ) -> None:
@@ -231,7 +231,7 @@ class CustomInstall(install):
         self.mpi = False
         self.hdfs = False
         self.precompile = False
-        self.nomp = False
+        self.nomp = True
         self.bit32 = False
 
     def run(self) -> None:
@@ -274,7 +274,7 @@ class CustomBdistWheel(bdist_wheel):
         self.mpi = False
         self.hdfs = False
         self.precompile = False
-        self.nomp = False
+        self.nomp = True
         self.bit32 = False
 
     def finalize_options(self) -> None:
-- 
2.17.1

