diff --git a/scipy/linalg/src/id_dist/src/idd_svd.c b/../../f2cfixes/idd_svd.c
index 75dfdcd..7020740 100644
--- a/scipy/linalg/src/id_dist/src/idd_svd.c
+++ b/../../f2cfixes/idd_svd.c
@@ -49,11 +49,10 @@
     static integer iftranspose, info;
     static char jobz[1];
     static integer ldvt;
-    extern /* Subroutine */ int idd_permuter__(integer *, doublereal *, 
-	    integer *, integer *, doublereal *);
+    extern /* Subroutine */ int idd_permuter__(integer *, integer *, integer *, integer *, doublereal *);
     static integer lwork;
     extern /* Subroutine */ int idd_retriever__(integer *, integer *, 
-	    doublereal *, integer *, doublereal *), dgesdd_(char *, integer *,
+	    doublereal *, integer *, doublereal *), wdgesdd_(char *, integer *,
 	     integer *, doublereal *, integer *, doublereal *, doublereal *, 
 	    integer *, doublereal *, integer *, doublereal *, integer *, 
 	    doublereal *, integer *, ftnlen);
@@ -139,7 +138,7 @@
     ldu = *krank;
     ldvt = *krank;
 
-    dgesdd_(jobz, krank, n, &r__[io + 1], &ldr, &s[1], &r__[io + *krank * *n 
+    wdgesdd_(jobz, krank, n, &r__[io + 1], &ldr, &s[1], &r__[io + *krank * *n 
 	    + 1], &ldu, &v[1], &ldvt, &r__[io + *krank * *n + *krank * *krank 
 	    + 1], &lwork, &r__[1], &info, (ftnlen)1);
 
@@ -214,11 +213,10 @@
     static integer iftranspose, info;
     static char jobz[1];
     static integer ldvt;
-    extern /* Subroutine */ int idd_permuter__(integer *, doublereal *, 
-	    integer *, integer *, doublereal *);
+    extern /* Subroutine */ int idd_permuter__(integer *, integer *, integer *, integer *, doublereal *);
     static integer lwork;
     extern /* Subroutine */ int idd_retriever__(integer *, integer *, 
-	    doublereal *, integer *, doublereal *), dgesdd_(char *, integer *,
+	    doublereal *, integer *, doublereal *), wdgesdd_(char *, integer *,
 	     integer *, doublereal *, integer *, doublereal *, doublereal *, 
 	    integer *, doublereal *, integer *, doublereal *, integer *, 
 	    doublereal *, integer *, ftnlen);
@@ -326,7 +324,7 @@
 	    return 0;
 	}
 
-	dgesdd_(jobz, krank, n, &w[io + 1], &ldr, &w[isi], &w[io + *krank * *
+	wdgesdd_(jobz, krank, n, &w[io + 1], &ldr, &w[isi], &w[io + *krank * *
 		n + 1], &ldu, &w[ivi], &ldvt, &w[io + *krank * *n + *krank * *
 		krank + 1], &lwork, &w[1], &info, (ftnlen)1);
 
@@ -405,8 +403,7 @@
 
 
 
-/* Subroutine */ int idd_permuter__(integer *krank, integer *ind, integer *m, 
-	integer *n, doublereal *a)
+/* Subroutine */ int idd_permuter__(integer *krank, integer *ind, integer *m, integer *n, doublereal *a)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1;
