diff --git a/scipy/linalg/src/id_dist/src/iddp_rsvd.c b/src/packages/scipy/f2cpatches/iddp_rsvd.c
index fec7f8b..5dce6a9 100644
--- a/scipy/linalg/src/id_dist/src/iddp_rsvd.c
+++ b/src/packages/scipy/f2cpatches/iddp_rsvd.c
@@ -40,11 +40,8 @@
 	     integer *, U_fp, doublereal *, doublereal *, doublereal *, 
 	    doublereal *, integer *, doublereal *, doublereal *, integer *);
     static integer k;
-    extern /* Subroutine */ int iddp_rsvd0__(integer *, integer *, U_fp, 
-	    doublereal *, doublereal *, doublereal *, doublereal *, U_fp, 
-	    doublereal *, doublereal *, doublereal *, doublereal *, integer *,
-	     doublereal *, doublereal *, doublereal *, integer *, doublereal *
-	    , doublereal *, doublereal *, doublereal *);
+    extern /* Subroutine */ int 
+iddp_rsvd0__(integer *, integer *, U_fp, doublereal *, doublereal *, doublereal *, doublereal *, U_fp, doublereal *, doublereal *, doublereal *, doublereal *, integer *,doublereal *, doublereal *, doublereal *, integer *, integer *, doublereal *, doublereal *, doublereal *);
     static integer lp, ls, lu, lv, lw2, isi, iui, ivi, icol, lcol, iproj, 
 	    ilist, lproj, llist, iwork, lwork;
 
@@ -232,12 +229,8 @@
 
 
 
-/* Subroutine */ int iddp_rsvd0__(integer *m, integer *n, U_fp matvect, 
-	doublereal *p1t, doublereal *p2t, doublereal *p3t, doublereal *p4t, 
-	U_fp matvec, doublereal *p1, doublereal *p2, doublereal *p3, 
-	doublereal *p4, integer *krank, doublereal *u, doublereal *v, 
-	doublereal *s, integer *ier, integer *list, doublereal *proj, 
-	doublereal *col, doublereal *work)
+/* Subroutine */ int 
+iddp_rsvd0__(integer *m, integer *n, U_fp matvect, doublereal *p1t, doublereal *p2t, doublereal *p3t, doublereal *p4t, U_fp matvec, doublereal *p1, doublereal *p2, doublereal *p3, doublereal *p4, integer *krank, doublereal *u, doublereal *v, doublereal *s, integer *ier, integer *list, doublereal *proj, doublereal *col, doublereal *work)
 {
     /* System generated locals */
     integer u_dim1, u_offset, v_dim1, v_offset, proj_dim1, proj_offset, i__1;
