diff --git a/scipy/linalg/src/id_dist/src/iddr_rsvd.c b/src/packages/scipy/f2cpatches/iddr_rsvd.c
index 260ff31..d134f43 100644
--- a/scipy/linalg/src/id_dist/src/iddr_rsvd.c
+++ b/src/packages/scipy/f2cpatches/iddr_rsvd.c
@@ -36,11 +36,8 @@
     integer u_dim1, u_offset, v_dim1, v_offset, i__1;
 
     /* Local variables */
-    extern /* Subroutine */ int iddr_rsvd0__(integer *, integer *, U_fp, 
-	    doublereal *, doublereal *, doublereal *, doublereal *, U_fp, 
-	    doublereal *, doublereal *, doublereal *, doublereal *, integer *,
-	     doublereal *, doublereal *, doublereal *, integer *, doublereal *
-	    , doublereal *, doublereal *, doublereal *);
+    extern /* Subroutine */ int 
+iddr_rsvd0__(integer *, integer *, U_fp, doublereal *, doublereal *, doublereal *, doublereal *, U_fp, doublereal *, doublereal *, doublereal *, doublereal *, integer *,doublereal *, doublereal *, doublereal *, integer *, integer *, doublereal *, doublereal *, doublereal *);
     static integer lw, icol, lcol, iproj, ilist, lproj, llist, iwork, lwork;
 
 
@@ -154,12 +151,8 @@
 
 
 
-/* Subroutine */ int iddr_rsvd0__(integer *m, integer *n, U_fp matvect, 
-	doublereal *p1t, doublereal *p2t, doublereal *p3t, doublereal *p4t, 
-	U_fp matvec, doublereal *p1, doublereal *p2, doublereal *p3, 
-	doublereal *p4, integer *krank, doublereal *u, doublereal *v, 
-	doublereal *s, integer *ier, integer *list, doublereal *proj, 
-	doublereal *col, doublereal *work)
+/* Subroutine */ int 
+iddr_rsvd0__(integer *m, integer *n, U_fp matvect, doublereal *p1t, doublereal *p2t, doublereal *p3t, doublereal *p4t, U_fp matvec, doublereal *p1, doublereal *p2, doublereal *p3, doublereal *p4, integer *krank, doublereal *u, doublereal *v, doublereal *s, integer *ier, integer *list, doublereal *proj, doublereal *col, doublereal *work)
 {
     /* System generated locals */
     integer u_dim1, u_offset, v_dim1, v_offset, i__1;
