diff --git a/scipy/linalg/src/id_dist/src/iddr_asvd.c b/../../f2cfixes/iddr_asvd.c
index 8f8095d..424dd83 100644
--- a/scipy/linalg/src/id_dist/src/iddr_asvd.c
+++ b/../../f2cfixes/iddr_asvd.c
@@ -32,10 +32,8 @@
     integer a_dim1, a_offset, u_dim1, u_offset, v_dim1, v_offset, i__1;
 
     /* Local variables */
-    extern /* Subroutine */ int iddr_asvd0__(integer *, integer *, doublereal 
-	    *, integer *, doublereal *, doublereal *, doublereal *, 
-	    doublereal *, integer *, doublereal *, doublereal *, doublereal *,
-	     doublereal *);
+    extern /* Subroutine */ int 
+iddr_asvd0__(integer *, integer *, doublereal *, integer *, doublereal *, doublereal *, doublereal *, doublereal *, integer *, integer *, doublereal *, doublereal *,doublereal *);
     static integer lw, icol, lcol, iproj, ilist, lproj, llist, iwork, lwork, 
 	    iwinit, lwinit;
 
@@ -122,10 +120,8 @@
 
 
 
-/* Subroutine */ int iddr_asvd0__(integer *m, integer *n, doublereal *a, 
-	integer *krank, doublereal *winit, doublereal *u, doublereal *v, 
-	doublereal *s, integer *ier, integer *list, doublereal *proj, 
-	doublereal *col, doublereal *work)
+/* Subroutine */ int 
+iddr_asvd0__(integer *m, integer *n, doublereal *a, integer *krank, doublereal *winit, doublereal *u, doublereal *v, doublereal *s, integer *ier, integer *list, doublereal *proj, doublereal *col, doublereal *work)
 {
     /* System generated locals */
     integer a_dim1, a_offset, u_dim1, u_offset, v_dim1, v_offset, proj_dim1, 
