diff --git a/scipy/linalg/src/id_dist/src/idz_frm.c b/../../f2cfixes/idz_frm.c
index 1fff7d2..b3db66d 100644
--- a/scipy/linalg/src/id_dist/src/idz_frm.c
+++ b/../../f2cfixes/idz_frm.c
@@ -47,10 +47,10 @@ static integer c__1 = 1;
     extern /* Subroutine */ int idz_random_transf__(doublecomplex *, 
 	    doublecomplex *, doublecomplex *);
     static integer iw;
-    extern /* Subroutine */ int idz_permute__(integer *, doublecomplex *, 
-	    doublecomplex *, doublecomplex *), zfftf_(integer *, 
-	    doublecomplex *, doublecomplex *), idz_subselect__(integer *, 
-	    doublecomplex *, integer *, doublecomplex *, doublecomplex *);
+    extern /* Subroutine */ int 
+idz_permute__(integer *, integer *, doublecomplex *, doublecomplex *), 
+zfftf_(integer *, doublecomplex *, doublecomplex *), 
+idz_subselect__(integer *, integer *, integer *, doublecomplex *, doublecomplex *);
 
 
 /*       transforms x into y via a composition */
@@ -139,8 +139,8 @@ static integer c__1 = 1;
 	    integer *, doublecomplex *, doublecomplex *), idz_random_transf__(
 	    doublecomplex *, doublecomplex *, doublecomplex *);
     static integer iw;
-    extern /* Subroutine */ int idz_subselect__(integer *, doublecomplex *, 
-	    integer *, doublecomplex *, doublecomplex *);
+    extern /* Subroutine */ int 
+idz_subselect__(integer *, integer *, integer *, doublecomplex *, doublecomplex *);
 
 
 /*       transforms x into y via a composition */
@@ -208,8 +208,8 @@ static integer c__1 = 1;
 
 
 
-/* Subroutine */ int idz_permute__(integer *n, integer *ind, doublecomplex *x,
-	 doublecomplex *y)
+/* Subroutine */ int 
+idz_permute__(integer *n, integer *ind, doublecomplex *x, doublecomplex *y)
 {
     /* System generated locals */
     integer i__1, i__2, i__3;
@@ -253,8 +253,8 @@ static integer c__1 = 1;
 
 
 
-/* Subroutine */ int idz_subselect__(integer *n, integer *ind, integer *m, 
-	doublecomplex *x, doublecomplex *y)
+/* Subroutine */ int 
+idz_subselect__(integer *n, integer *ind, integer *m, doublecomplex *x, doublecomplex *y)
 {
     /* System generated locals */
     integer i__1, i__2, i__3;
