diff --git a/scipy/linalg/src/id_dist/src/idz_id2svd.c b/../../f2cfixes/idz_id2svd.c
index e6cec1e..8ddcc74 100644
--- a/scipy/linalg/src/id_dist/src/idz_id2svd.c
+++ b/../../f2cfixes/idz_id2svd.c
@@ -34,12 +34,8 @@
 
     /* Local variables */
     static integer ip, ir, lp, it, lr, lt, lw;
-    extern /* Subroutine */ int idz_id2svd0__(integer *, integer *, 
-	    doublecomplex *, integer *, integer *, doublecomplex *, 
-	    doublecomplex *, doublecomplex *, doublereal *, integer *, 
-	    doublecomplex *, doublecomplex *, doublecomplex *, doublecomplex *
-	    , doublecomplex *, doublecomplex *, doublecomplex *, 
-	    doublecomplex *);
+    extern /* Subroutine */ int 
+idz_id2svd0__(integer *, integer *, doublecomplex *, integer *, integer *, doublecomplex *, doublecomplex *, doublecomplex *, doublereal *, integer *, doublecomplex *, doublecomplex *, doublecomplex *, doublecomplex *, doublecomplex *, doublecomplex *, integer *, integer *);
     static integer ir2, ir3, lr2, lr3, iind, lind, iindt, lindt, iwork, lwork;
 
 
@@ -141,11 +137,8 @@
 
 
 
-/* Subroutine */ int idz_id2svd0__(integer *m, integer *krank, doublecomplex *
-	b, integer *n, integer *list, doublecomplex *proj, doublecomplex *u, 
-	doublecomplex *v, doublereal *s, integer *ier, doublecomplex *work, 
-	doublecomplex *p, doublecomplex *t, doublecomplex *r__, doublecomplex 
-	*r2, doublecomplex *r3, integer *ind, integer *indt)
+/* Subroutine */ int 
+idz_id2svd0__(integer *m, integer *krank, doublecomplex *b, integer *n, integer *list, doublecomplex *proj, doublecomplex *u, doublecomplex *v, doublereal *s, integer *ier, doublecomplex *work, doublecomplex *p, doublecomplex *t, doublecomplex *r__, doublecomplex *r2, doublecomplex *r3, integer *ind, integer *indt)
 {
     /* System generated locals */
     integer b_dim1, b_offset, proj_dim1, proj_offset, p_dim1, p_offset, 
@@ -173,7 +166,7 @@
 	    integer *, doublecomplex *, integer *, doublecomplex *, 
 	    doublecomplex *);
     static integer lwork;
-    extern /* Subroutine */ int zgesdd_(char *, integer *, integer *, 
+    extern /* Subroutine */ int wzgesdd_(char *, integer *, integer *, 
 	    doublecomplex *, integer *, doublereal *, doublecomplex *, 
 	    integer *, doublecomplex *, integer *, doublecomplex *, integer *,
 	     doublecomplex *, doublecomplex *, integer *, ftnlen);
@@ -301,7 +294,7 @@
     i__3 = *krank;
 /* Computing 2nd power */
     i__4 = *krank;
-    zgesdd_(jobz, krank, krank, &r3[r3_offset], &ldr, &s[1], &work[1], &ldu, &
+    wzgesdd_(jobz, krank, krank, &r3[r3_offset], &ldr, &s[1], &work[1], &ldu, &
 	    r__[r_offset], &ldvt, &work[i__1 * i__1 + (*krank << 1) + i__2 * 
 	    i__2 * 3 + (*krank << 2) + 1], &lwork, &work[i__3 * i__3 + (*
 	    krank << 1) + 1], &work[i__4 * i__4 + 1], &info, (ftnlen)1);
