diff --git a/scipy/linalg/src/id_dist/src/idz_svd.c b/../../f2cfixes/idz_svd.c
index 2f0a73a..d387c73 100644
--- a/scipy/linalg/src/id_dist/src/idz_svd.c
+++ b/../../f2cfixes/idz_svd.c
@@ -51,13 +51,13 @@
 	    doublecomplex *);
     static integer info;
     static char jobz[1];
-    extern /* Subroutine */ int idz_permuter__(integer *, doublecomplex *, 
-	    integer *, integer *, doublecomplex *);
+    extern /* Subroutine */ int 
+idz_permuter__(integer *, integer *, integer *, integer *, doublecomplex *);
     static integer lwork;
     extern /* Subroutine */ int idz_retriever__(integer *, integer *, 
 	    doublecomplex *, integer *, doublecomplex *);
     static integer ldvadj;
-    extern /* Subroutine */ int zgesdd_(char *, integer *, integer *, 
+    extern /* Subroutine */ int wzgesdd_(char *, integer *, integer *, 
 	    doublecomplex *, integer *, doublereal *, doublecomplex *, 
 	    integer *, doublecomplex *, integer *, doublecomplex *, integer *,
 	     doublecomplex *, doublecomplex *, integer *, ftnlen);
@@ -141,7 +141,7 @@
     ldu = *krank;
     ldvadj = *krank;
 
-    zgesdd_(jobz, krank, n, &r__[io + 1], &ldr, &s[1], &r__[io + *krank * *n 
+    wzgesdd_(jobz, krank, n, &r__[io + 1], &ldr, &s[1], &r__[io + *krank * *n 
 	    + 1], &ldu, &v[1], &ldvadj, &r__[io + *krank * *n + *krank * *
 	    krank + 1], &lwork, &r__[io + *krank * *n + *krank * *krank + 
 	    lwork + 1], &r__[1], &info, (ftnlen)1);
@@ -224,14 +224,14 @@
 	    doublecomplex *);
     static integer info;
     static char jobz[1];
-    extern /* Subroutine */ int idz_realcomp__(integer *, doublecomplex *, 
-	    doublecomplex *), idz_permuter__(integer *, doublecomplex *, 
-	    integer *, integer *, doublecomplex *);
+    extern /* Subroutine */ int 
+idz_realcomp__(integer *, doublereal *, doublecomplex *), 
+idz_permuter__(integer *, integer *, integer *, integer *, doublecomplex *);
     static integer lwork;
     extern /* Subroutine */ int idz_retriever__(integer *, integer *, 
 	    doublecomplex *, integer *, doublecomplex *);
     static integer ldvadj;
-    extern /* Subroutine */ int zgesdd_();
+    extern /* Subroutine */ int wzgesdd_();
 
 
 /*       constructs a rank-krank SVD  U Sigma V^*  approximating a */
@@ -337,7 +337,7 @@
 	    return 0;
 	}
 
-	zgesdd_(jobz, krank, n, &w[io + 1], &ldr, &w[isi], &w[io + *krank * *
+	wzgesdd_(jobz, krank, n, &w[io + 1], &ldr, &w[isi], &w[io + *krank * *
 		n + 1], &ldu, &w[ivi], &ldvadj, &w[io + *krank * *n + *krank *
 		 *krank + 1], &lwork, &w[io + *krank * *n + *krank * *krank + 
 		lwork + 1], &w[1], &info, (ftnlen)1);
@@ -416,8 +416,8 @@
 
 
 
-/* Subroutine */ int idz_realcomp__(integer *n, doublereal *a, doublecomplex *
-	b)
+/* Subroutine */ int 
+idz_realcomp__(integer *n, doublereal *a, doublecomplex *b)
 {
     /* System generated locals */
     integer i__1, i__2, i__3;
@@ -458,8 +458,8 @@
 
 
 
-/* Subroutine */ int idz_permuter__(integer *krank, integer *ind, integer *m, 
-	integer *n, doublecomplex *a)
+/* Subroutine */ int 
+idz_permuter__(integer *krank, integer *ind, integer *m, integer *n, doublecomplex *a)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1, i__2, i__3;
