diff --git a/scipy/linalg/src/id_dist/src/idzp_aid.c b/../../f2cfixes/idzp_aid.c
index 40b3f24..d56b61f 100644
--- a/scipy/linalg/src/id_dist/src/idzp_aid.c
+++ b/../../f2cfixes/idzp_aid.c
@@ -36,11 +36,9 @@
     integer a_dim1, a_offset;
 
     /* Local variables */
-    extern /* Subroutine */ int idzp_aid0__(doublereal *, integer *, integer *
-	    , doublecomplex *, integer *, integer *, doublecomplex *, 
-	    doublecomplex *), idzp_aid1__(doublereal *, integer *, integer *, 
-	    integer *, doublecomplex *, integer *, integer *, doublecomplex *)
-	    ;
+    extern /* Subroutine */ int 
+idzp_aid0__(doublereal *, integer *, integer *, doublecomplex *, integer *, integer *, doublecomplex *, doublereal *), 
+idzp_aid1__(doublereal *, integer *, integer *, integer *, doublecomplex *, integer *, integer *, doublereal *);
     static integer n2;
     extern /* Subroutine */ int idz_estrank__(doublereal *, integer *, 
 	    integer *, doublecomplex *, doublecomplex *, integer *, 
@@ -139,9 +137,8 @@
 
 
 
-/* Subroutine */ int idzp_aid0__(doublereal *eps, integer *m, integer *n, 
-	doublecomplex *a, integer *krank, integer *list, doublecomplex *proj, 
-	doublereal *rnorms)
+/* Subroutine */ int 
+idzp_aid0__(doublereal *eps, integer *m, integer *n, doublecomplex *a, integer *krank, integer *list, doublecomplex *proj, doublereal *rnorms)
 {
     /* System generated locals */
     integer a_dim1, a_offset, proj_dim1, proj_offset, i__1, i__2, i__3, i__4;
@@ -213,9 +210,8 @@
 
 
 
-/* Subroutine */ int idzp_aid1__(doublereal *eps, integer *n2, integer *n, 
-	integer *kranki, doublecomplex *proj, integer *krank, integer *list, 
-	doublereal *rnorms)
+/* Subroutine */ int 
+idzp_aid1__(doublereal *eps, integer *n2, integer *n, integer *kranki, doublecomplex *proj, integer *krank, integer *list, doublereal *rnorms)
 {
     /* System generated locals */
     integer i__1, i__2, i__3, i__4;
@@ -290,9 +286,8 @@
 
     /* Local variables */
     static integer n2, ira, lra, lra2;
-    extern /* Subroutine */ int idz_estrank0__(doublereal *, integer *, 
-	    integer *, doublecomplex *, doublecomplex *, integer *, integer *,
-	     doublecomplex *, doublecomplex *, doublecomplex *);
+    extern /* Subroutine */ int 
+idz_estrank0__(doublereal *, integer *, integer *, doublecomplex *, doublecomplex *, integer *, integer *,doublecomplex *, doublecomplex *, doublereal *);
     static integer irat, lrat, iscal, lscal;
 
 
@@ -374,9 +369,8 @@
 
 
 
-/* Subroutine */ int idz_estrank0__(doublereal *eps, integer *m, integer *n, 
-	doublecomplex *a, doublecomplex *w, integer *n2, integer *krank, 
-	doublecomplex *ra, doublecomplex *rat, doublereal *scal)
+/* Subroutine */ int 
+idz_estrank0__(doublereal *eps, integer *m, integer *n, doublecomplex *a, doublecomplex *w, integer *n2, integer *krank, doublecomplex *ra, doublecomplex *rat, doublereal *scal)
 {
     /* System generated locals */
     integer a_dim1, a_offset, ra_dim1, ra_offset, rat_dim1, rat_offset, i__1, 
