diff --git a/scipy/linalg/src/id_dist/src/idzp_rsvd.c b/src/packages/scipy/f2cpatches/idzp_rsvd.c
index ea140ec..ed4e57a 100644
--- a/scipy/linalg/src/id_dist/src/idzp_rsvd.c
+++ b/src/packages/scipy/f2cpatches/idzp_rsvd.c
@@ -37,18 +37,16 @@
     integer i__1, i__2, i__3;
 
     /* Local variables */
-    extern /* Subroutine */ int idz_reco__(integer *, doublecomplex *, 
-	    doublecomplex *), idzp_rid__(integer *, doublereal *, integer *, 
+    extern /* Subroutine */ int 
+idz_reco__(integer *, doublereal *, doublecomplex *), 
+idzp_rid__(integer *, doublereal *, integer *, 
 	    integer *, U_fp, doublecomplex *, doublecomplex *, doublecomplex *
 	    , doublecomplex *, integer *, doublecomplex *, doublecomplex *, 
 	    integer *);
     static integer k, lp, ls, lu, lv;
-    extern /* Subroutine */ int idzp_rsvd0__(integer *, integer *, U_fp, 
-	    doublecomplex *, doublecomplex *, doublecomplex *, doublecomplex *
-	    , U_fp, doublecomplex *, doublecomplex *, doublecomplex *, 
-	    doublecomplex *, integer *, doublecomplex *, doublecomplex *, 
-	    doublecomplex *, integer *, doublecomplex *, doublecomplex *, 
-	    doublecomplex *, doublecomplex *);
+    extern /* Subroutine */ 
+int 
+idzp_rsvd0__(integer *, integer *, U_fp, doublecomplex *, doublecomplex *, doublecomplex *, doublecomplex *, U_fp, doublecomplex *, doublecomplex *, doublecomplex *, doublecomplex *, integer *, doublecomplex *, doublecomplex *, doublereal *, integer *, integer *, doublecomplex *, doublecomplex *, doublecomplex *);
     static integer lw2, isi, iui, ivi, icol, lcol, iproj, ilist, lproj, llist,
 	     iwork, lwork;
 
@@ -237,12 +235,8 @@
 
 
 
-/* Subroutine */ int idzp_rsvd0__(integer *m, integer *n, U_fp matveca, 
-	doublecomplex *p1t, doublecomplex *p2t, doublecomplex *p3t, 
-	doublecomplex *p4t, U_fp matvec, doublecomplex *p1, doublecomplex *p2,
-	 doublecomplex *p3, doublecomplex *p4, integer *krank, doublecomplex *
-	u, doublecomplex *v, doublereal *s, integer *ier, integer *list, 
-	doublecomplex *proj, doublecomplex *col, doublecomplex *work)
+/* Subroutine */ int 
+idzp_rsvd0__(integer *m, integer *n, U_fp matveca, doublecomplex *p1t, doublecomplex *p2t, doublecomplex *p3t, doublecomplex *p4t, U_fp matvec, doublecomplex *p1, doublecomplex *p2,doublecomplex *p3, doublecomplex *p4, integer *krank, doublecomplex *	u, doublecomplex *v, doublereal *s, integer *ier, integer *list, doublecomplex *proj, doublecomplex *col, doublecomplex *work)
 {
     /* System generated locals */
     integer u_dim1, u_offset, v_dim1, v_offset, proj_dim1, proj_offset, i__1;
@@ -299,7 +293,8 @@
 
 
 
-/* Subroutine */ int idz_reco__(integer *n, doublereal *a, doublecomplex *b)
+/* Subroutine */ int 
+idz_reco__(integer *n, doublereal *a, doublecomplex *b)
 {
     /* System generated locals */
     integer i__1, i__2, i__3;
