diff --git a/scipy/linalg/src/id_dist/src/idzr_asvd.c b/../../f2cfixes/idzr_asvd.c
index 490ccae..3ed1b7e 100644
--- a/scipy/linalg/src/id_dist/src/idzr_asvd.c
+++ b/../../f2cfixes/idzr_asvd.c
@@ -32,10 +32,8 @@
     integer a_dim1, a_offset, u_dim1, u_offset, v_dim1, v_offset, i__1;
 
     /* Local variables */
-    extern /* Subroutine */ int idzr_asvd0__(integer *, integer *, 
-	    doublecomplex *, integer *, doublecomplex *, doublecomplex *, 
-	    doublecomplex *, doublereal *, integer *, doublecomplex *, 
-	    doublecomplex *, doublecomplex *, doublecomplex *);
+    extern /* Subroutine */ int 
+idzr_asvd0__(integer *, integer *, doublecomplex *, integer *, doublecomplex *, doublecomplex *, doublecomplex *, doublereal *, integer *, integer *, doublecomplex *, doublecomplex *, doublecomplex *);
     static integer lw, icol, lcol, iproj, ilist, lproj, llist, iwork, lwork, 
 	    iwinit, lwinit;
 
@@ -123,10 +121,8 @@
 
 
 
-/* Subroutine */ int idzr_asvd0__(integer *m, integer *n, doublecomplex *a, 
-	integer *krank, doublecomplex *winit, doublecomplex *u, doublecomplex 
-	*v, doublereal *s, integer *ier, integer *list, doublecomplex *proj, 
-	doublecomplex *col, doublecomplex *work)
+/* Subroutine */ int 
+idzr_asvd0__(integer *m, integer *n, doublecomplex *a, integer *krank, doublecomplex *winit, doublecomplex *u, doublecomplex *v, doublereal *s, integer *ier, integer *list, doublecomplex *proj, doublecomplex *col, doublecomplex *work)
 {
     /* System generated locals */
     integer a_dim1, a_offset, u_dim1, u_offset, v_dim1, v_offset, proj_dim1, 
