diff --git a/scipy/linalg/src/id_dist/src/idzr_rsvd.c b/../../f2cfixes/idzr_rsvd.c
index ce0c8d1..bd914c4 100644
--- a/scipy/linalg/src/id_dist/src/idzr_rsvd.c
+++ b/../../f2cfixes/idzr_rsvd.c
@@ -37,12 +37,8 @@
 
     /* Local variables */
     static integer lw;
-    extern /* Subroutine */ int idzr_rsvd0__(integer *, integer *, U_fp, 
-	    doublecomplex *, doublecomplex *, doublecomplex *, doublecomplex *
-	    , U_fp, doublecomplex *, doublecomplex *, doublecomplex *, 
-	    doublecomplex *, integer *, doublecomplex *, doublecomplex *, 
-	    doublereal *, integer *, doublecomplex *, doublecomplex *, 
-	    doublecomplex *, doublecomplex *);
+    extern /* Subroutine */ int 
+    idzr_rsvd0__(integer *, integer *, U_fp, doublecomplex *, doublecomplex *, doublecomplex *, doublecomplex *, U_fp, doublecomplex *, doublecomplex *, doublecomplex *, doublecomplex *, integer *, doublecomplex *, doublecomplex *, doublereal *, integer *, integer *, doublecomplex *, doublecomplex *, doublecomplex *);
     static integer icol, lcol, iproj, ilist, lproj, llist, iwork, lwork;
 
 
@@ -156,12 +152,8 @@
 
 
 
-/* Subroutine */ int idzr_rsvd0__(integer *m, integer *n, U_fp matveca, 
-	doublecomplex *p1t, doublecomplex *p2t, doublecomplex *p3t, 
-	doublecomplex *p4t, U_fp matvec, doublecomplex *p1, doublecomplex *p2,
-	 doublecomplex *p3, doublecomplex *p4, integer *krank, doublecomplex *
-	u, doublecomplex *v, doublereal *s, integer *ier, integer *list, 
-	doublecomplex *proj, doublecomplex *col, doublecomplex *work)
+/* Subroutine */ int 
+idzr_rsvd0__(integer *m, integer *n, U_fp matveca, doublecomplex *p1t, doublecomplex *p2t, doublecomplex *p3t, doublecomplex *p4t, U_fp matvec, doublecomplex *p1, doublecomplex *p2,doublecomplex *p3, doublecomplex *p4, integer *krank, doublecomplex *u, doublecomplex *v, doublereal *s, integer *ier, integer *list, doublecomplex *proj, doublecomplex *col, doublecomplex *work)
 {
     /* System generated locals */
     integer u_dim1, u_offset, v_dim1, v_offset, i__1, i__2, i__3;
