diff --git a/wrap_dummy_g77_abi.bak.c b/wrap_dummy_g77_abi.c
index f25b5bd..37c79c6 100644
--- a/wrap_dummy_g77_abi.bak.c
+++ b/wrap_dummy_g77_abi.c
@@ -251,10 +251,7 @@ static integer c__2 = 2;
 /* > \ingroup complexGEcomputational */
 
 /*  ===================================================================== */
-/* Subroutine */ int cgemqrt_(char *side, char *trans, integer *m, integer *n,
-	 integer *k, integer *nb, complex *v, integer *ldv, complex *t, 
-	integer *ldt, complex *c__, integer *ldc, complex *work, integer *
-	info, ftnlen side_len, ftnlen trans_len)
+/* Subroutine */ int cgemqrt_(char *side, char *trans, integer *m, integer *n, integer *k, integer *nb, complex *v, integer *ldv, complex *t, integer *ldt, complex *c__, integer *ldc, complex *work, integer * info)
 {
     /* System generated locals */
     integer v_dim1, v_offset, c_dim1, c_offset, t_dim1, t_offset, i__1, i__2, 
@@ -573,8 +570,7 @@ static integer c__2 = 2;
 /* > \endverbatim */
 /* > */
 /*  ===================================================================== */
-/* Subroutine */ int cgeqrfp_(integer *m, integer *n, complex *a, integer *
-	lda, complex *tau, complex *work, integer *lwork, integer *info)
+/* Subroutine */ int cgeqrfp_(integer *m, integer *n, complex *a, integer * lda, complex *tau, complex *work, integer *lwork, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1, i__2, i__3, i__4;
@@ -889,8 +885,7 @@ f"> */
 /* > \endverbatim */
 /* > */
 /*  ===================================================================== */
-/* Subroutine */ int cgeqrt_(integer *m, integer *n, integer *nb, complex *a, 
-	integer *lda, complex *t, integer *ldt, complex *work, integer *info)
+/* Subroutine */ int cgeqrt_(integer *m, integer *n, integer *nb, complex *a, integer *lda, complex *t, integer *ldt, complex *work, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, t_dim1, t_offset, i__1, i__2, i__3, i__4, i__5;
@@ -1115,9 +1110,7 @@ f"> */
 /* > \ingroup complexSYcomputational */
 
 /*  ===================================================================== */
-/* Subroutine */ int csyconv_(char *uplo, char *way, integer *n, complex *a, 
-	integer *lda, integer *ipiv, complex *e, integer *info, ftnlen 
-	uplo_len, ftnlen way_len)
+/* Subroutine */ int csyconv_(char *uplo, char *way, integer *n, complex *a, integer *lda, integer *ipiv, complex *e, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1, i__2, i__3;
@@ -1645,9 +1638,7 @@ f.f"> */
 /* > */
 /* > \endverbatim */
 /*  ===================================================================== */
-/* Subroutine */ int csyconvf_(char *uplo, char *way, integer *n, complex *a, 
-	integer *lda, complex *e, integer *ipiv, integer *info, ftnlen 
-	uplo_len, ftnlen way_len)
+/* Subroutine */ int csyconvf_(char *uplo, char *way, integer *n, complex *a, integer *lda, complex *e, integer *ipiv, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1, i__2;
@@ -2234,9 +2225,7 @@ f_rook.f"> */
 /* > */
 /* > \endverbatim */
 /*  ===================================================================== */
-/* Subroutine */ int csyconvf_rook__(char *uplo, char *way, integer *n, 
-	complex *a, integer *lda, complex *e, integer *ipiv, integer *info, 
-	ftnlen uplo_len, ftnlen way_len)
+/* Subroutine */ int csyconvf_rook__(char *uplo, char *way, integer *n, complex *a, integer *lda, complex *e, integer *ipiv, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1, i__2;
@@ -2836,11 +2825,7 @@ f_rook.f"> */
 /* > \endverbatim */
 /* > */
 /*  ===================================================================== */
-/* Subroutine */ int ctpmqrt_(char *side, char *trans, integer *m, integer *n,
-	 integer *k, integer *l, integer *nb, complex *v, integer *ldv, 
-	complex *t, integer *ldt, complex *a, integer *lda, complex *b, 
-	integer *ldb, complex *work, integer *info, ftnlen side_len, ftnlen 
-	trans_len)
+/* Subroutine */ int ctpmqrt_(char *side, char *trans, integer *m, integer *n, integer *k, integer *l, integer *nb, complex *v, integer *ldv, complex *t, integer *ldt, complex *a, integer *lda, complex *b, integer *ldb, complex *work, integer *info)
 {
     /* System generated locals */
     integer v_dim1, v_offset, a_dim1, a_offset, b_dim1, b_offset, t_dim1, 
@@ -2855,8 +2840,7 @@ f_rook.f"> */
     extern /* Subroutine */ int xerbla_(char *, integer *, ftnlen), ctprfb_(
 	    char *, char *, char *, char *, integer *, integer *, integer *, 
 	    integer *, complex *, integer *, complex *, integer *, complex *, 
-	    integer *, complex *, integer *, complex *, integer *, ftnlen, 
-	    ftnlen, ftnlen, ftnlen);
+	    integer *, complex *, integer *, complex *, integer *);
     static logical notran;
 
 
@@ -2967,8 +2951,7 @@ f_rook.f"> */
 	    }
 	    ctprfb_("L", "C", "F", "C", &mb, n, &ib, &lb, &v[i__ * v_dim1 + 1]
 		    , ldv, &t[i__ * t_dim1 + 1], ldt, &a[i__ + a_dim1], lda, &
-		    b[b_offset], ldb, &work[1], &ib, (ftnlen)1, (ftnlen)1, (
-		    ftnlen)1, (ftnlen)1);
+		    b[b_offset], ldb, &work[1], &ib);
 	}
 
     } else if (right && notran) {
@@ -2989,8 +2972,7 @@ f_rook.f"> */
 	    }
 	    ctprfb_("R", "N", "F", "C", m, &mb, &ib, &lb, &v[i__ * v_dim1 + 1]
 		    , ldv, &t[i__ * t_dim1 + 1], ldt, &a[i__ * a_dim1 + 1], 
-		    lda, &b[b_offset], ldb, &work[1], m, (ftnlen)1, (ftnlen)1,
-		     (ftnlen)1, (ftnlen)1);
+		    lda, &b[b_offset], ldb, &work[1], m);
 	}
 
     } else if (left && notran) {
@@ -3011,8 +2993,7 @@ f_rook.f"> */
 	    }
 	    ctprfb_("L", "N", "F", "C", &mb, n, &ib, &lb, &v[i__ * v_dim1 + 1]
 		    , ldv, &t[i__ * t_dim1 + 1], ldt, &a[i__ + a_dim1], lda, &
-		    b[b_offset], ldb, &work[1], &ib, (ftnlen)1, (ftnlen)1, (
-		    ftnlen)1, (ftnlen)1);
+		    b[b_offset], ldb, &work[1], &ib);
 	}
 
     } else if (right && tran) {
@@ -3033,8 +3014,7 @@ f_rook.f"> */
 	    }
 	    ctprfb_("R", "C", "F", "C", m, &mb, &ib, &lb, &v[i__ * v_dim1 + 1]
 		    , ldv, &t[i__ * t_dim1 + 1], ldt, &a[i__ * a_dim1 + 1], 
-		    lda, &b[b_offset], ldb, &work[1], m, (ftnlen)1, (ftnlen)1,
-		     (ftnlen)1, (ftnlen)1);
+		    lda, &b[b_offset], ldb, &work[1], m);
 	}
 
     }
@@ -3234,9 +3214,7 @@ f"> */
 /* > \endverbatim */
 /* > */
 /*  ===================================================================== */
-/* Subroutine */ int ctpqrt_(integer *m, integer *n, integer *l, integer *nb, 
-	complex *a, integer *lda, complex *b, integer *ldb, complex *t, 
-	integer *ldt, complex *work, integer *info)
+/* Subroutine */ int ctpqrt_(integer *m, integer *n, integer *l, integer *nb, complex *a, integer *lda, complex *b, integer *ldb, complex *t, integer *ldt, complex *work, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, b_dim1, b_offset, t_dim1, t_offset, i__1, i__2, 
@@ -3247,8 +3225,7 @@ f"> */
     extern /* Subroutine */ int xerbla_(char *, integer *, ftnlen), ctprfb_(
 	    char *, char *, char *, char *, integer *, integer *, integer *, 
 	    integer *, complex *, integer *, complex *, integer *, complex *, 
-	    integer *, complex *, integer *, complex *, integer *, ftnlen, 
-	    ftnlen, ftnlen, ftnlen), ctpqrt2_(integer *, integer *, integer *,
+	    integer *, complex *, integer *, complex *, integer *), ctpqrt2_(integer *, integer *, integer *,
 	     complex *, integer *, complex *, integer *, complex *, integer *,
 	     integer *);
 
@@ -3342,7 +3319,7 @@ f"> */
 	    ctprfb_("L", "C", "F", "C", &mb, &i__3, &ib, &lb, &b[i__ * b_dim1 
 		    + 1], ldb, &t[i__ * t_dim1 + 1], ldt, &a[i__ + (i__ + ib) 
 		    * a_dim1], lda, &b[(i__ + ib) * b_dim1 + 1], ldb, &work[1]
-		    , &ib, (ftnlen)1, (ftnlen)1, (ftnlen)1, (ftnlen)1);
+		    , &ib);
 	}
     }
     return 0;
@@ -3519,10 +3496,7 @@ f"> */
 /* > \ingroup doubleGEcomputational */
 
 /*  ===================================================================== */
-/* Subroutine */ int dgemqrt_(char *side, char *trans, integer *m, integer *n,
-	 integer *k, integer *nb, doublereal *v, integer *ldv, doublereal *t, 
-	integer *ldt, doublereal *c__, integer *ldc, doublereal *work, 
-	integer *info, ftnlen side_len, ftnlen trans_len)
+/* Subroutine */ int dgemqrt_(char *side, char *trans, integer *m, integer *n, integer *k, integer *nb, doublereal *v, integer *ldv, doublereal *t, integer *ldt, doublereal *c__, integer *ldc, doublereal *work, integer *info)
 {
     /* System generated locals */
     integer v_dim1, v_offset, c_dim1, c_offset, t_dim1, t_offset, i__1, i__2, 
@@ -3842,8 +3816,7 @@ f"> */
 /* > \endverbatim */
 /* > */
 /*  ===================================================================== */
-/* Subroutine */ int dgeqrfp_(integer *m, integer *n, doublereal *a, integer *
-	lda, doublereal *tau, doublereal *work, integer *lwork, integer *info)
+/* Subroutine */ int dgeqrfp_(integer *m, integer *n, doublereal *a, integer * lda, doublereal *tau, doublereal *work, integer *lwork, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1, i__2, i__3, i__4;
@@ -4159,9 +4132,7 @@ f"> */
 /* > \endverbatim */
 /* > */
 /*  ===================================================================== */
-/* Subroutine */ int dgeqrt_(integer *m, integer *n, integer *nb, doublereal *
-	a, integer *lda, doublereal *t, integer *ldt, doublereal *work, 
-	integer *info)
+/* Subroutine */ int dgeqrt_(integer *m, integer *n, integer *nb, doublereal * a, integer *lda, doublereal *t, integer *ldt, doublereal *work, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, t_dim1, t_offset, i__1, i__2, i__3, i__4, i__5;
@@ -4387,9 +4358,7 @@ f"> */
 /* > \ingroup doubleSYcomputational */
 
 /*  ===================================================================== */
-/* Subroutine */ int dsyconv_(char *uplo, char *way, integer *n, doublereal *
-	a, integer *lda, integer *ipiv, doublereal *e, integer *info, ftnlen 
-	uplo_len, ftnlen way_len)
+/* Subroutine */ int dsyconv_(char *uplo, char *way, integer *n, doublereal * a, integer *lda, integer *ipiv, doublereal *e, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1;
@@ -4867,9 +4836,7 @@ f.f"> */
 /* > */
 /* > \endverbatim */
 /*  ===================================================================== */
-/* Subroutine */ int dsyconvf_(char *uplo, char *way, integer *n, doublereal *
-	a, integer *lda, doublereal *e, integer *ipiv, integer *info, ftnlen 
-	uplo_len, ftnlen way_len)
+/* Subroutine */ int dsyconvf_(char *uplo, char *way, integer *n, doublereal * a, integer *lda, doublereal *e, integer *ipiv, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1;
@@ -5438,9 +5405,7 @@ f_rook.f"> */
 /* > */
 /* > \endverbatim */
 /*  ===================================================================== */
-/* Subroutine */ int dsyconvf_rook__(char *uplo, char *way, integer *n, 
-	doublereal *a, integer *lda, doublereal *e, integer *ipiv, integer *
-	info, ftnlen uplo_len, ftnlen way_len)
+/* Subroutine */ int dsyconvf_rook__(char *uplo, char *way, integer *n, doublereal *a, integer *lda, doublereal *e, integer *ipiv, integer * info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1;
@@ -6025,11 +5990,7 @@ f_rook.f"> */
 /* > \endverbatim */
 /* > */
 /*  ===================================================================== */
-/* Subroutine */ int dtpmqrt_(char *side, char *trans, integer *m, integer *n,
-	 integer *k, integer *l, integer *nb, doublereal *v, integer *ldv, 
-	doublereal *t, integer *ldt, doublereal *a, integer *lda, doublereal *
-	b, integer *ldb, doublereal *work, integer *info, ftnlen side_len, 
-	ftnlen trans_len)
+/* Subroutine */ int dtpmqrt_(char *side, char *trans, integer *m, integer *n, integer *k, integer *l, integer *nb, doublereal *v, integer *ldv, doublereal *t, integer *ldt, doublereal *a, integer *lda, doublereal * b, integer *ldb, doublereal *work, integer *info)
 {
     /* System generated locals */
     integer v_dim1, v_offset, a_dim1, a_offset, b_dim1, b_offset, t_dim1, 
@@ -6041,11 +6002,13 @@ f_rook.f"> */
     static integer ldvq;
     extern logical lsame_(char *, char *, ftnlen, ftnlen);
     static logical right;
-    extern /* Subroutine */ int xerbla_(char *, integer *, ftnlen), dtprfb_(
+    extern /* Subroutine */ int xerbla_(char *, integer *, ftnlen), 
+	
+	dtprfb_(
 	    char *, char *, char *, char *, integer *, integer *, integer *, 
 	    integer *, doublereal *, integer *, doublereal *, integer *, 
 	    doublereal *, integer *, doublereal *, integer *, doublereal *, 
-	    integer *, ftnlen, ftnlen, ftnlen, ftnlen);
+	    integer *);
     static logical notran;
 
 
@@ -6156,8 +6119,7 @@ f_rook.f"> */
 	    }
 	    dtprfb_("L", "T", "F", "C", &mb, n, &ib, &lb, &v[i__ * v_dim1 + 1]
 		    , ldv, &t[i__ * t_dim1 + 1], ldt, &a[i__ + a_dim1], lda, &
-		    b[b_offset], ldb, &work[1], &ib, (ftnlen)1, (ftnlen)1, (
-		    ftnlen)1, (ftnlen)1);
+		    b[b_offset], ldb, &work[1], &ib);
 	}
 
     } else if (right && notran) {
@@ -6178,8 +6140,7 @@ f_rook.f"> */
 	    }
 	    dtprfb_("R", "N", "F", "C", m, &mb, &ib, &lb, &v[i__ * v_dim1 + 1]
 		    , ldv, &t[i__ * t_dim1 + 1], ldt, &a[i__ * a_dim1 + 1], 
-		    lda, &b[b_offset], ldb, &work[1], m, (ftnlen)1, (ftnlen)1,
-		     (ftnlen)1, (ftnlen)1);
+		    lda, &b[b_offset], ldb, &work[1], m);
 	}
 
     } else if (left && notran) {
@@ -6200,8 +6161,7 @@ f_rook.f"> */
 	    }
 	    dtprfb_("L", "N", "F", "C", &mb, n, &ib, &lb, &v[i__ * v_dim1 + 1]
 		    , ldv, &t[i__ * t_dim1 + 1], ldt, &a[i__ + a_dim1], lda, &
-		    b[b_offset], ldb, &work[1], &ib, (ftnlen)1, (ftnlen)1, (
-		    ftnlen)1, (ftnlen)1);
+		    b[b_offset], ldb, &work[1], &ib);
 	}
 
     } else if (right && tran) {
@@ -6222,8 +6182,7 @@ f_rook.f"> */
 	    }
 	    dtprfb_("R", "T", "F", "C", m, &mb, &ib, &lb, &v[i__ * v_dim1 + 1]
 		    , ldv, &t[i__ * t_dim1 + 1], ldt, &a[i__ * a_dim1 + 1], 
-		    lda, &b[b_offset], ldb, &work[1], m, (ftnlen)1, (ftnlen)1,
-		     (ftnlen)1, (ftnlen)1);
+		    lda, &b[b_offset], ldb, &work[1], m);
 	}
 
     }
@@ -6423,9 +6382,7 @@ f"> */
 /* > \endverbatim */
 /* > */
 /*  ===================================================================== */
-/* Subroutine */ int dtpqrt_(integer *m, integer *n, integer *l, integer *nb, 
-	doublereal *a, integer *lda, doublereal *b, integer *ldb, doublereal *
-	t, integer *ldt, doublereal *work, integer *info)
+/* Subroutine */ int dtpqrt_(integer *m, integer *n, integer *l, integer *nb, doublereal *a, integer *lda, doublereal *b, integer *ldb, doublereal * t, integer *ldt, doublereal *work, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, b_dim1, b_offset, t_dim1, t_offset, i__1, i__2, 
@@ -6433,13 +6390,12 @@ f"> */
 
     /* Local variables */
     static integer i__, ib, lb, mb, iinfo;
-    extern /* Subroutine */ int xerbla_(char *, integer *, ftnlen), dtprfb_(
-	    char *, char *, char *, char *, integer *, integer *, integer *, 
-	    integer *, doublereal *, integer *, doublereal *, integer *, 
-	    doublereal *, integer *, doublereal *, integer *, doublereal *, 
-	    integer *, ftnlen, ftnlen, ftnlen, ftnlen), dtpqrt2_(integer *, 
-	    integer *, integer *, doublereal *, integer *, doublereal *, 
-	    integer *, doublereal *, integer *, integer *);
+    extern /* Subroutine */ int xerbla_(char *, integer *, ftnlen), 
+	
+dtprfb_(char *, char *, char *, char *, integer *, integer *, integer *, integer *, doublereal *, integer *, doublereal *, integer *, 
+	    doublereal *, integer *, doublereal *, integer *, doublereal *, integer *), 
+		
+		dtpqrt2_(integer *, integer *, integer *, doublereal *, integer *, doublereal *, integer *, doublereal *, integer *, integer *);
 
 
 /*  -- LAPACK computational routine -- */
@@ -6531,9 +6487,8 @@ f"> */
 	    dtprfb_("L", "T", "F", "C", &mb, &i__3, &ib, &lb, &b[i__ * b_dim1 
 		    + 1], ldb, &t[i__ * t_dim1 + 1], ldt, &a[i__ + (i__ + ib) 
 		    * a_dim1], lda, &b[(i__ + ib) * b_dim1 + 1], ldb, &work[1]
-		    , &ib, (ftnlen)1, (ftnlen)1, (ftnlen)1, (ftnlen)1);
-	}
-    }
+		    , &ib);
+	}}
     return 0;
 
 /*     End of DTPQRT */
@@ -6708,10 +6663,7 @@ f"> */
 /* > \ingroup realGEcomputational */
 
 /*  ===================================================================== */
-/* Subroutine */ int sgemqrt_(char *side, char *trans, integer *m, integer *n,
-	 integer *k, integer *nb, real *v, integer *ldv, real *t, integer *
-	ldt, real *c__, integer *ldc, real *work, integer *info, ftnlen 
-	side_len, ftnlen trans_len)
+/* Subroutine */ int sgemqrt_(char *side, char *trans, integer *m, integer *n, integer *k, integer *nb, real *v, integer *ldv, real *t, integer * ldt, real *c__, integer *ldc, real *work, integer *info)
 {
     /* System generated locals */
     integer v_dim1, v_offset, c_dim1, c_offset, t_dim1, t_offset, i__1, i__2, 
@@ -7030,8 +6982,7 @@ f"> */
 /* > \endverbatim */
 /* > */
 /*  ===================================================================== */
-/* Subroutine */ int sgeqrfp_(integer *m, integer *n, real *a, integer *lda, 
-	real *tau, real *work, integer *lwork, integer *info)
+/* Subroutine */ int sgeqrfp_(integer *m, integer *n, real *a, integer *lda, real *tau, real *work, integer *lwork, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1, i__2, i__3, i__4;
@@ -7346,8 +7297,7 @@ f"> */
 /* > \endverbatim */
 /* > */
 /*  ===================================================================== */
-/* Subroutine */ int sgeqrt_(integer *m, integer *n, integer *nb, real *a, 
-	integer *lda, real *t, integer *ldt, real *work, integer *info)
+/* Subroutine */ int sgeqrt_(integer *m, integer *n, integer *nb, real *a, integer *lda, real *t, integer *ldt, real *work, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, t_dim1, t_offset, i__1, i__2, i__3, i__4, i__5;
@@ -7572,9 +7522,7 @@ f"> */
 /* > \ingroup realSYcomputational */
 
 /*  ===================================================================== */
-/* Subroutine */ int ssyconv_(char *uplo, char *way, integer *n, real *a, 
-	integer *lda, integer *ipiv, real *e, integer *info, ftnlen uplo_len, 
-	ftnlen way_len)
+/* Subroutine */ int ssyconv_(char *uplo, char *way, integer *n, real *a, integer *lda, integer *ipiv, real *e, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1;
@@ -8052,9 +8000,7 @@ f.f"> */
 /* > */
 /* > \endverbatim */
 /*  ===================================================================== */
-/* Subroutine */ int ssyconvf_(char *uplo, char *way, integer *n, real *a, 
-	integer *lda, real *e, integer *ipiv, integer *info, ftnlen uplo_len, 
-	ftnlen way_len)
+/* Subroutine */ int ssyconvf_(char *uplo, char *way, integer *n, real *a, integer *lda, real *e, integer *ipiv, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1;
@@ -8622,9 +8568,7 @@ f_rook.f"> */
 /* > */
 /* > \endverbatim */
 /*  ===================================================================== */
-/* Subroutine */ int ssyconvf_rook__(char *uplo, char *way, integer *n, real *
-	a, integer *lda, real *e, integer *ipiv, integer *info, ftnlen 
-	uplo_len, ftnlen way_len)
+/* Subroutine */ int ssyconvf_rook__(char *uplo, char *way, integer *n, real * a, integer *lda, real *e, integer *ipiv, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1;
@@ -9208,10 +9152,7 @@ f_rook.f"> */
 /* > \endverbatim */
 /* > */
 /*  ===================================================================== */
-/* Subroutine */ int stpmqrt_(char *side, char *trans, integer *m, integer *n,
-	 integer *k, integer *l, integer *nb, real *v, integer *ldv, real *t, 
-	integer *ldt, real *a, integer *lda, real *b, integer *ldb, real *
-	work, integer *info, ftnlen side_len, ftnlen trans_len)
+/* Subroutine */ int stpmqrt_(char *side, char *trans, integer *m, integer *n, integer *k, integer *l, integer *nb, real *v, integer *ldv, real *t, integer *ldt, real *a, integer *lda, real *b, integer *ldb, real * work, integer *info)
 {
     /* System generated locals */
     integer v_dim1, v_offset, a_dim1, a_offset, b_dim1, b_offset, t_dim1, 
@@ -9226,8 +9167,7 @@ f_rook.f"> */
     extern /* Subroutine */ int xerbla_(char *, integer *, ftnlen), stprfb_(
 	    char *, char *, char *, char *, integer *, integer *, integer *, 
 	    integer *, real *, integer *, real *, integer *, real *, integer *
-	    , real *, integer *, real *, integer *, ftnlen, ftnlen, ftnlen, 
-	    ftnlen);
+	    , real *, integer *, real *, integer *);
     static logical notran;
 
 
@@ -9338,8 +9278,7 @@ f_rook.f"> */
 	    }
 	    stprfb_("L", "T", "F", "C", &mb, n, &ib, &lb, &v[i__ * v_dim1 + 1]
 		    , ldv, &t[i__ * t_dim1 + 1], ldt, &a[i__ + a_dim1], lda, &
-		    b[b_offset], ldb, &work[1], &ib, (ftnlen)1, (ftnlen)1, (
-		    ftnlen)1, (ftnlen)1);
+		    b[b_offset], ldb, &work[1], &ib);
 	}
 
     } else if (right && notran) {
@@ -9360,8 +9299,7 @@ f_rook.f"> */
 	    }
 	    stprfb_("R", "N", "F", "C", m, &mb, &ib, &lb, &v[i__ * v_dim1 + 1]
 		    , ldv, &t[i__ * t_dim1 + 1], ldt, &a[i__ * a_dim1 + 1], 
-		    lda, &b[b_offset], ldb, &work[1], m, (ftnlen)1, (ftnlen)1,
-		     (ftnlen)1, (ftnlen)1);
+		    lda, &b[b_offset], ldb, &work[1], m);
 	}
 
     } else if (left && notran) {
@@ -9382,8 +9320,7 @@ f_rook.f"> */
 	    }
 	    stprfb_("L", "N", "F", "C", &mb, n, &ib, &lb, &v[i__ * v_dim1 + 1]
 		    , ldv, &t[i__ * t_dim1 + 1], ldt, &a[i__ + a_dim1], lda, &
-		    b[b_offset], ldb, &work[1], &ib, (ftnlen)1, (ftnlen)1, (
-		    ftnlen)1, (ftnlen)1);
+		    b[b_offset], ldb, &work[1], &ib);
 	}
 
     } else if (right && tran) {
@@ -9404,8 +9341,7 @@ f_rook.f"> */
 	    }
 	    stprfb_("R", "T", "F", "C", m, &mb, &ib, &lb, &v[i__ * v_dim1 + 1]
 		    , ldv, &t[i__ * t_dim1 + 1], ldt, &a[i__ * a_dim1 + 1], 
-		    lda, &b[b_offset], ldb, &work[1], m, (ftnlen)1, (ftnlen)1,
-		     (ftnlen)1, (ftnlen)1);
+		    lda, &b[b_offset], ldb, &work[1], m);
 	}
 
     }
@@ -9605,9 +9541,7 @@ f"> */
 /* > \endverbatim */
 /* > */
 /*  ===================================================================== */
-/* Subroutine */ int stpqrt_(integer *m, integer *n, integer *l, integer *nb, 
-	real *a, integer *lda, real *b, integer *ldb, real *t, integer *ldt, 
-	real *work, integer *info)
+/* Subroutine */ int stpqrt_(integer *m, integer *n, integer *l, integer *nb, real *a, integer *lda, real *b, integer *ldb, real *t, integer *ldt, real *work, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, b_dim1, b_offset, t_dim1, t_offset, i__1, i__2, 
@@ -9618,8 +9552,7 @@ f"> */
     extern /* Subroutine */ int xerbla_(char *, integer *, ftnlen), stprfb_(
 	    char *, char *, char *, char *, integer *, integer *, integer *, 
 	    integer *, real *, integer *, real *, integer *, real *, integer *
-	    , real *, integer *, real *, integer *, ftnlen, ftnlen, ftnlen, 
-	    ftnlen), stpqrt2_(integer *, integer *, integer *, real *, 
+	    , real *, integer *, real *, integer *), stpqrt2_(integer *, integer *, integer *, real *, 
 	    integer *, real *, integer *, real *, integer *, integer *);
 
 
@@ -9712,7 +9645,7 @@ f"> */
 	    stprfb_("L", "T", "F", "C", &mb, &i__3, &ib, &lb, &b[i__ * b_dim1 
 		    + 1], ldb, &t[i__ * t_dim1 + 1], ldt, &a[i__ + (i__ + ib) 
 		    * a_dim1], lda, &b[(i__ + ib) * b_dim1 + 1], ldb, &work[1]
-		    , &ib, (ftnlen)1, (ftnlen)1, (ftnlen)1, (ftnlen)1);
+		    , &ib);
 	}
     }
     return 0;
@@ -9889,10 +9822,7 @@ f"> */
 /* > \ingroup complex16GEcomputational */
 
 /*  ===================================================================== */
-/* Subroutine */ int zgemqrt_(char *side, char *trans, integer *m, integer *n,
-	 integer *k, integer *nb, doublecomplex *v, integer *ldv, 
-	doublecomplex *t, integer *ldt, doublecomplex *c__, integer *ldc, 
-	doublecomplex *work, integer *info, ftnlen side_len, ftnlen trans_len)
+/* Subroutine */ int zgemqrt_(char *side, char *trans, integer *m, integer *n, integer *k, integer *nb, doublecomplex *v, integer *ldv, doublecomplex *t, integer *ldt, doublecomplex *c__, integer *ldc, doublecomplex *work, integer *info)
 {
     /* System generated locals */
     integer v_dim1, v_offset, c_dim1, c_offset, t_dim1, t_offset, i__1, i__2, 
@@ -10212,9 +10142,7 @@ f"> */
 /* > \endverbatim */
 /* > */
 /*  ===================================================================== */
-/* Subroutine */ int zgeqrfp_(integer *m, integer *n, doublecomplex *a, 
-	integer *lda, doublecomplex *tau, doublecomplex *work, integer *lwork,
-	 integer *info)
+/* Subroutine */ int zgeqrfp_(integer *m, integer *n, doublecomplex *a, integer *lda, doublecomplex *tau, doublecomplex *work, integer *lwork, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1, i__2, i__3, i__4;
@@ -10532,9 +10460,7 @@ f"> */
 /* > \endverbatim */
 /* > */
 /*  ===================================================================== */
-/* Subroutine */ int zgeqrt_(integer *m, integer *n, integer *nb, 
-	doublecomplex *a, integer *lda, doublecomplex *t, integer *ldt, 
-	doublecomplex *work, integer *info)
+/* Subroutine */ int zgeqrt_(integer *m, integer *n, integer *nb, doublecomplex *a, integer *lda, doublecomplex *t, integer *ldt, doublecomplex *work, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, t_dim1, t_offset, i__1, i__2, i__3, i__4, i__5;
@@ -10760,9 +10686,7 @@ f"> */
 /* > \ingroup complex16SYcomputational */
 
 /*  ===================================================================== */
-/* Subroutine */ int zsyconv_(char *uplo, char *way, integer *n, 
-	doublecomplex *a, integer *lda, integer *ipiv, doublecomplex *e, 
-	integer *info, ftnlen uplo_len, ftnlen way_len)
+/* Subroutine */ int zsyconv_(char *uplo, char *way, integer *n, doublecomplex *a, integer *lda, integer *ipiv, doublecomplex *e, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1, i__2, i__3;
@@ -11292,9 +11216,7 @@ f.f"> */
 /* > */
 /* > \endverbatim */
 /*  ===================================================================== */
-/* Subroutine */ int zsyconvf_(char *uplo, char *way, integer *n, 
-	doublecomplex *a, integer *lda, doublecomplex *e, integer *ipiv, 
-	integer *info, ftnlen uplo_len, ftnlen way_len)
+/* Subroutine */ int zsyconvf_(char *uplo, char *way, integer *n, doublecomplex *a, integer *lda, doublecomplex *e, integer *ipiv, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1, i__2;
@@ -11880,9 +11802,7 @@ f_rook.f"> */
 /* > */
 /* > \endverbatim */
 /*  ===================================================================== */
-/* Subroutine */ int zsyconvf_rook__(char *uplo, char *way, integer *n, 
-	doublecomplex *a, integer *lda, doublecomplex *e, integer *ipiv, 
-	integer *info, ftnlen uplo_len, ftnlen way_len)
+/* Subroutine */ int zsyconvf_rook__(char *uplo, char *way, integer *n, doublecomplex *a, integer *lda, doublecomplex *e, integer *ipiv, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, i__1, i__2;
@@ -12481,11 +12401,7 @@ f_rook.f"> */
 /* > \endverbatim */
 /* > */
 /*  ===================================================================== */
-/* Subroutine */ int ztpmqrt_(char *side, char *trans, integer *m, integer *n,
-	 integer *k, integer *l, integer *nb, doublecomplex *v, integer *ldv, 
-	doublecomplex *t, integer *ldt, doublecomplex *a, integer *lda, 
-	doublecomplex *b, integer *ldb, doublecomplex *work, integer *info, 
-	ftnlen side_len, ftnlen trans_len)
+/* Subroutine */ int ztpmqrt_(char *side, char *trans, integer *m, integer *n, integer *k, integer *l, integer *nb, doublecomplex *v, integer *ldv, doublecomplex *t, integer *ldt, doublecomplex *a, integer *lda, doublecomplex *b, integer *ldb, doublecomplex *work, integer *info)
 {
     /* System generated locals */
     integer v_dim1, v_offset, a_dim1, a_offset, b_dim1, b_offset, t_dim1, 
@@ -12502,8 +12418,7 @@ f_rook.f"> */
     extern /* Subroutine */ int ztprfb_(char *, char *, char *, char *, 
 	    integer *, integer *, integer *, integer *, doublecomplex *, 
 	    integer *, doublecomplex *, integer *, doublecomplex *, integer *,
-	     doublecomplex *, integer *, doublecomplex *, integer *, ftnlen, 
-	    ftnlen, ftnlen, ftnlen);
+	     doublecomplex *, integer *, doublecomplex *, integer *);
 
 
 /*  -- LAPACK computational routine -- */
@@ -12613,8 +12528,7 @@ f_rook.f"> */
 	    }
 	    ztprfb_("L", "C", "F", "C", &mb, n, &ib, &lb, &v[i__ * v_dim1 + 1]
 		    , ldv, &t[i__ * t_dim1 + 1], ldt, &a[i__ + a_dim1], lda, &
-		    b[b_offset], ldb, &work[1], &ib, (ftnlen)1, (ftnlen)1, (
-		    ftnlen)1, (ftnlen)1);
+		    b[b_offset], ldb, &work[1], &ib);
 	}
 
     } else if (right && notran) {
@@ -12635,8 +12549,7 @@ f_rook.f"> */
 	    }
 	    ztprfb_("R", "N", "F", "C", m, &mb, &ib, &lb, &v[i__ * v_dim1 + 1]
 		    , ldv, &t[i__ * t_dim1 + 1], ldt, &a[i__ * a_dim1 + 1], 
-		    lda, &b[b_offset], ldb, &work[1], m, (ftnlen)1, (ftnlen)1,
-		     (ftnlen)1, (ftnlen)1);
+		    lda, &b[b_offset], ldb, &work[1], m);
 	}
 
     } else if (left && notran) {
@@ -12657,8 +12570,7 @@ f_rook.f"> */
 	    }
 	    ztprfb_("L", "N", "F", "C", &mb, n, &ib, &lb, &v[i__ * v_dim1 + 1]
 		    , ldv, &t[i__ * t_dim1 + 1], ldt, &a[i__ + a_dim1], lda, &
-		    b[b_offset], ldb, &work[1], &ib, (ftnlen)1, (ftnlen)1, (
-		    ftnlen)1, (ftnlen)1);
+		    b[b_offset], ldb, &work[1], &ib);
 	}
 
     } else if (right && tran) {
@@ -12679,8 +12591,7 @@ f_rook.f"> */
 	    }
 	    ztprfb_("R", "C", "F", "C", m, &mb, &ib, &lb, &v[i__ * v_dim1 + 1]
 		    , ldv, &t[i__ * t_dim1 + 1], ldt, &a[i__ * a_dim1 + 1], 
-		    lda, &b[b_offset], ldb, &work[1], m, (ftnlen)1, (ftnlen)1,
-		     (ftnlen)1, (ftnlen)1);
+		    lda, &b[b_offset], ldb, &work[1], m);
 	}
 
     }
@@ -12880,9 +12791,7 @@ f"> */
 /* > \endverbatim */
 /* > */
 /*  ===================================================================== */
-/* Subroutine */ int ztpqrt_(integer *m, integer *n, integer *l, integer *nb, 
-	doublecomplex *a, integer *lda, doublecomplex *b, integer *ldb, 
-	doublecomplex *t, integer *ldt, doublecomplex *work, integer *info)
+/* Subroutine */ int ztpqrt_(integer *m, integer *n, integer *l, integer *nb, doublecomplex *a, integer *lda, doublecomplex *b, integer *ldb, doublecomplex *t, integer *ldt, doublecomplex *work, integer *info)
 {
     /* System generated locals */
     integer a_dim1, a_offset, b_dim1, b_offset, t_dim1, t_offset, i__1, i__2, 
@@ -12894,7 +12803,7 @@ f"> */
 	    char *, char *, char *, char *, integer *, integer *, integer *, 
 	    integer *, doublecomplex *, integer *, doublecomplex *, integer *,
 	     doublecomplex *, integer *, doublecomplex *, integer *, 
-	    doublecomplex *, integer *, ftnlen, ftnlen, ftnlen, ftnlen), 
+	    doublecomplex *, integer *), 
 	    ztpqrt2_(integer *, integer *, integer *, doublecomplex *, 
 	    integer *, doublecomplex *, integer *, doublecomplex *, integer *,
 	     integer *);
@@ -12989,7 +12898,7 @@ f"> */
 	    ztprfb_("L", "C", "F", "C", &mb, &i__3, &ib, &lb, &b[i__ * b_dim1 
 		    + 1], ldb, &t[i__ * t_dim1 + 1], ldt, &a[i__ + (i__ + ib) 
 		    * a_dim1], lda, &b[(i__ + ib) * b_dim1 + 1], ldb, &work[1]
-		    , &ib, (ftnlen)1, (ftnlen)1, (ftnlen)1, (ftnlen)1);
+		    , &ib);
 	}
     }
     return 0;
@@ -12998,3 +12907,10 @@ f"> */
 
 } /* ztpqrt_ */
 
+
+typedef struct { float real, imag; } npy_complex64;
+typedef struct { double real, imag; } npy_complex128;
+int sorcsd_(char *jobu1, char *jobu2, char *jobv1t, char *jobv2t, char *trans, char *signs, int *m, int *p, int *q, float *x11, int *ldx11, float *x12, int *ldx12, float *x21, int *ldx21, float *x22, int *ldx22, float *theta, float *u1, int *ldu1, float *u2, int *ldu2, float *v1t, int *ldv1t, float *v2t, int *ldv2t, float *work, int *lwork, int *iwork, int *info){}
+int dorcsd_(char *jobu1, char *jobu2, char *jobv1t, char *jobv2t, char *trans, char *signs, int *m, int *p, int *q, double *x11, int *ldx11, double *x12, int *ldx12, double *x21, int *ldx21, double *x22, int *ldx22, double *theta, double *u1, int *ldu1, double *u2, int *ldu2, double *v1t, int *ldv1t, double *v2t, int *ldv2t, double *work, int *lwork, int *iwork, int *info){}
+int zuncsd_(char *jobu1, char *jobu2, char *jobv1t, char *jobv2t, char *trans, char *signs, int *m, int *p, int *q, npy_complex128 *x11, int *ldx11, npy_complex128 *x12, int *ldx12, npy_complex128 *x21, int *ldx21, npy_complex128 *x22, int *ldx22, double *theta, npy_complex128 *u1, int *ldu1, npy_complex128 *u2, int *ldu2, npy_complex128 *v1t, int *ldv1t, npy_complex128 *v2t, int *ldv2t, npy_complex128 *work, int *lwork, double *rwork, int *lrwork, int *iwork, int *info){}
+int cuncsd_(char *jobu1, char *jobu2, char *jobv1t, char *jobv2t, char *trans, char *signs, int *m, int *p, int *q, npy_complex64 *x11, int *ldx11, npy_complex64 *x12, int *ldx12, npy_complex64 *x21, int *ldx21, npy_complex64 *x22, int *ldx22, float *theta, npy_complex64 *u1, int *ldu1, npy_complex64 *u2, int *ldu2, npy_complex64 *v1t, int *ldv1t, npy_complex64 *v2t, int *ldv2t, npy_complex64 *work, int *lwork, float *rwork, int *lrwork, int *iwork, int *info){}
