From 247456c9c94278c57d5103df9bed9e3eade95d7c Mon Sep 17 00:00:00 2001
From: Hood <hood@mit.edu>
Date: Wed, 7 Jul 2021 09:24:48 -0700
Subject: [PATCH] Hard code _PyEval_EvalFrame to call _PyEval_EvalFrameDefault

---
 Include/internal/pycore_ceval.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Include/internal/pycore_ceval.h b/Include/internal/pycore_ceval.h
index 18c8f02..ddb3d22 100644
--- a/Include/internal/pycore_ceval.h
+++ b/Include/internal/pycore_ceval.h
@@ -37,7 +37,7 @@ void _PyEval_Fini(void);
 static inline PyObject*
 _PyEval_EvalFrame(PyThreadState *tstate, PyFrameObject *f, int throwflag)
 {
-    return tstate->interp->eval_frame(tstate, f, throwflag);
+    return _PyEval_EvalFrameDefault(tstate, f, throwflag);
 }
 
 extern PyObject *_PyEval_EvalCode(
-- 
2.17.1

