From f78bb95d3c95a4fb20745eb70ad876495de26e94 Mon Sep 17 00:00:00 2001
From: Michael Droettboom <mdboom@gmail.com>
Date: Sun, 5 Jul 2020 17:37:43 +0200
Subject: [PATCH] disable-set_inheritable

---
 Python/fileutils.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Python/fileutils.c b/Python/fileutils.c
index 769ab59..a784789 100644
--- a/Python/fileutils.c
+++ b/Python/fileutils.c
@@ -1134,6 +1134,9 @@ _Py_get_inheritable(int fd)
 static int
 set_inheritable(int fd, int inheritable, int raise, int *atomic_flag_works)
 {
+#ifdef EMSCRIPTEN
+    return 0;
+#endif
 #ifdef MS_WINDOWS
     HANDLE handle;
     DWORD flags;
-- 
2.17.1

