From ba3bbdfc19c8d1d778c36af0424c56e193460170 Mon Sep 17 00:00:00 2001
From: Michael Droettboom <mdboom@gmail.com>
Date: Sun, 5 Jul 2020 17:38:21 +0200
Subject: [PATCH] dont-test-undecodable-filenames

---
 Lib/test/support/__init__.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Lib/test/support/__init__.py b/Lib/test/support/__init__.py
index aee3737..d0be99f 100644
--- a/Lib/test/support/__init__.py
+++ b/Lib/test/support/__init__.py
@@ -809,6 +809,8 @@ if os.name == 'nt':
                   'Unicode filename tests may not be effective'
                   % (TESTFN_UNENCODABLE, TESTFN_ENCODING))
             TESTFN_UNENCODABLE = None
+elif sys.platform == 'emscripten':
+    pass
 # Mac OS X denies unencodable filenames (invalid utf-8)
 elif sys.platform != 'darwin':
     try:
-- 
2.17.1

