From 9217312c422eb2cfef0968a1101eb2aa7b87d22a Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Sun, 13 Nov 2022 09:35:59 -0800
Subject: [PATCH 2/3] `descr_set_trampoline_call` return type should be `int`
 not `PyObject*`

---
 Objects/descrobject.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Objects/descrobject.c b/Objects/descrobject.c
index c3c541bf3c..c1fa897e9a 100644
--- a/Objects/descrobject.c
+++ b/Objects/descrobject.c
@@ -16,7 +16,7 @@ class property "propertyobject *" "&PyProperty_Type"
 // see pycore_object.h
 #if defined(__EMSCRIPTEN__) && defined(PY_CALL_TRAMPOLINE)
 #include <emscripten.h>
-EM_JS(PyObject*, descr_set_trampoline_call, (setter set, PyObject *obj, PyObject *value, void *closure), {
+EM_JS(int, descr_set_trampoline_call, (setter set, PyObject *obj, PyObject *value, void *closure), {
     return wasmTable.get(set)(obj, value, closure);
 });
 
-- 
2.25.1

