From f8970422b395b24ca58cef3dca6954bb07d4974d Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Mon, 19 Dec 2022 09:09:14 -0800
Subject: [PATCH 4/8] Allow multiprocessing.connection top level import

---
 Lib/multiprocessing/connection.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Lib/multiprocessing/connection.py b/Lib/multiprocessing/connection.py
index b08144f7a1..6a98829169 100644
--- a/Lib/multiprocessing/connection.py
+++ b/Lib/multiprocessing/connection.py
@@ -18,7 +18,10 @@
 import tempfile
 import itertools
 
-import _multiprocessing
+try:
+    import _multiprocessing
+except ModuleNotFoundError:
+    pass
 
 from . import util
 
-- 
2.25.1

