---
 config.sub   | 9 ++++++++-
 configure    | 6 ++++++
 configure.ac | 5 +++++
 3 files changed, 19 insertions(+), 1 deletion(-)

diff --git a/config.sub b/config.sub
index ba37cf9..0d22b33 100755
--- a/config.sub
+++ b/config.sub
@@ -118,7 +118,8 @@ case $maybe_os in
   linux-musl* | linux-uclibc* | uclinux-uclibc* | uclinux-gnu* | kfreebsd*-gnu* | \
   knetbsd*-gnu* | netbsd*-gnu* | netbsd*-eabi* | \
   kopensolaris*-gnu* | cloudabi*-eabi* | \
-  storm-chaos* | os2-emx* | rtmk-nova*)
+  storm-chaos* | os2-emx* | rtmk-nova* | \
+  emscripten)
     os=-$maybe_os
     basic_machine=`echo "$1" | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\1/'`
     ;;
@@ -377,6 +378,7 @@ case $basic_machine in
 	| alpha64-* | alpha64ev[4-8]-* | alpha64ev56-* | alpha64ev6[78]-* \
 	| alphapca5[67]-* | alpha64pca5[67]-* | arc-* | arceb-* \
 	| arm-*  | armbe-* | armle-* | armeb-* | armv*-* \
+	| wasm32 \
 	| avr-* | avr32-* \
 	| ba-* \
 	| be32-* | be64-* \
@@ -527,6 +529,9 @@ case $basic_machine in
 	asmjs)
 		basic_machine=asmjs-unknown
 		;;
+	wasm32)
+		basic_machine=wasm32-unknown
+		;;
 	aux)
 		basic_machine=m68k-apple
 		os=-aux
@@ -1522,6 +1527,8 @@ case $os in
 			;;
 		esac
 		;;
+	-emscripten)
+		;;
 	-nacl*)
 		;;
 	-ios)
diff --git a/configure b/configure
index 8dcdbf1..e1195d2 100755
--- a/configure
+++ b/configure
@@ -3307,6 +3307,9 @@ then
 	*-*-vxworks*)
 	    ac_sys_system=VxWorks
 	    ;;
+	wasm32-*-*)
+		ac_sys_system=Emscripten
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -3357,6 +3360,9 @@ if test "$cross_compiling" = yes; then
 	*-*-vxworks*)
 		_host_cpu=$host_cpu
 		;;
+	wasm32-*-*)
+		_host_cpu=
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
diff --git a/configure.ac b/configure.ac
index b1e4c6c..dbfae6d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -403,6 +403,9 @@ then
 	*-*-vxworks*)
 	    ac_sys_system=VxWorks
 	    ;;
+	wasm32-*-*)
+		ac_sys_system=Emscripten
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -451,6 +454,8 @@ if test "$cross_compiling" = yes; then
 		;;
 	*-*-vxworks*)
 		_host_cpu=$host_cpu
+	wasm32-*-*)
+		_host_cpu=
 		;;
 	*)
 		# for now, limit cross builds to known configurations
-- 
2.17.1

