From 2119fa65d7af744fd0050214d3452312bbb8d063 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Tue, 22 Oct 2024 15:16:03 +0200
Subject: [PATCH 5/8] Add fallback trampoline for JSPI with no type reflection

---
 Python/emscripten_trampoline.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Python/emscripten_trampoline.c b/Python/emscripten_trampoline.c
index 8d29393bd87..812cc10167f 100644
--- a/Python/emscripten_trampoline.c
+++ b/Python/emscripten_trampoline.c
@@ -10,7 +10,7 @@
  * https://github.com/GoogleChromeLabs/wasm-feature-detect/blob/main/src/detectors/type-reflection/index.js
  */
 EM_JS(int, _PyEM_detect_type_reflection, (), {
-  return "Function" in WebAssembly;
+    return !!Module.PyEM_CountArgs;
 });
 
 void
@@ -36,7 +36,7 @@ EM_JS(int, _PyEM_CountFuncParams, (PyCFunctionWithKeywords func), {
   if (n !== undefined) {
     return n;
   }
-  n = wasmFunctionType(wasmTable.get(func)).parameters.length;
+  n = Module.PyEM_CountArgs(func);
   _PyEM_CountFuncParams.cache.set(func, n);
   return n;
 }
-- 
2.34.1

