From 4c71c808cc65ed6003b1e29d583c71586ebb36e1 Mon Sep 17 00:00:00 2001
From: ryanking13 <def6488@gmail.com>
Date: Wed, 25 Jan 2023 15:54:16 +0900
Subject: [PATCH 8/8] Move test directories

---
 Makefile.pre.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index b356f6293e..68c55a356a 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -1932,8 +1932,7 @@ LIBSUBDIRS=	asyncio \
 		xmlrpc \
 		zoneinfo \
 		__phello__
-TESTSUBDIRS=	ctypes/test \
-		distutils/tests \
+TESTSUBDIRS= distutils/tests \
 		idlelib/idle_test \
 		lib2to3/tests \
 		lib2to3/tests/data \
@@ -2009,7 +2008,8 @@ TESTSUBDIRS=	ctypes/test \
 		test/ziptestdata \
 		tkinter/test tkinter/test/test_tkinter \
 		tkinter/test/test_ttk \
-		unittest/test unittest/test/testmock
+		test/test_ctypes \
+		test/test_unittest test/test_unittest/testmock
 
 TEST_MODULES=@TEST_MODULES@
 libinstall:	all $(srcdir)/Modules/xxmodule.c
-- 
2.29.2.windows.2

