From 3d8d569dce3bca982b87a05efaaf3a8da6e4cd07 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Mon, 19 Jun 2023 12:32:05 -0700
Subject: [PATCH 8/8] Remove _PyCFunctionWithKeywords_TrampolineCall definition

---
 Objects/methodobject.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/Objects/methodobject.c b/Objects/methodobject.c
index 953cf4666d..969338d701 100644
--- a/Objects/methodobject.c
+++ b/Objects/methodobject.c
@@ -558,7 +558,4 @@ cfunction_call(PyObject *func, PyObject *args, PyObject *kwargs)
 #if defined(__EMSCRIPTEN__) && defined(PY_CALL_TRAMPOLINE)
 #include <emscripten.h>
 
-EM_JS(PyObject*, _PyCFunctionWithKeywords_TrampolineCall, (PyCFunctionWithKeywords func, PyObject *self, PyObject *args, PyObject *kw), {
-    return wasmTable.get(func)(self, args, kw);
-});
 #endif
-- 
2.25.1

