diff -ur matplotlib-2.2.2/lib/matplotlib/cbook/__init__.py matplotlib-2.2.2/lib/matplotlib/cbook/__init__.py
--- a/lib/matplotlib/cbook/__init__.py	2018-03-05 23:47:09.000000000 -0500
+++ b/lib/matplotlib/cbook/__init__.py	2018-05-17 09:39:11.514431317 -0400
@@ -10,7 +10,7 @@

 import six
 from six.moves import xrange, zip
-import bz2
+# import bz2
 import collections
 import contextlib
 import datetime
@@ -601,10 +601,10 @@
             # get rid of 'U' in flag for gzipped files.
             flag = flag.replace('U', '')
             fh = gzip.open(fname, flag)
-        elif fname.endswith('.bz2'):
-            # get rid of 'U' in flag for bz2 files
-            flag = flag.replace('U', '')
-            fh = bz2.BZ2File(fname, flag)
+        # elif fname.endswith('.bz2'):
+        #     # get rid of 'U' in flag for bz2 files
+        #     flag = flag.replace('U', '')
+        #     fh = bz2.BZ2File(fname, flag)
         else:
             fh = io.open(fname, flag, encoding=encoding)
         opened = True
