From 896618743d27517d75875e722ccc75e708714eeb Mon Sep 17 00:00:00 2001
From: Jo Bovy <bovy@astro.utoronto.ca>
Date: Mon, 25 Jul 2022 23:28:34 -0400
Subject: [PATCH 1/2] rand_r --> rand

---
 src/stochastic_forces.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/stochastic_forces.c b/src/stochastic_forces.c
index f67f18e..3c98ac7 100644
--- a/src/stochastic_forces.c
+++ b/src/stochastic_forces.c
@@ -65,8 +65,8 @@
 static void rebx_random_normal2(struct reb_simulation* r, double* n0, double* n1){
 	double v1,v2,rsq=1.;
 	while(rsq>=1. || rsq<1.0e-12){
-		v1=2.*((double)rand_r(&(r->rand_seed)))/((double)(RAND_MAX))-1.0;
-		v2=2.*((double)rand_r(&(r->rand_seed)))/((double)(RAND_MAX))-1.0;
+	        v1=2.*((double)rand())/((double)(RAND_MAX))-1.0;
+		v2=2.*((double)rand())/((double)(RAND_MAX))-1.0;
 		rsq=v1*v1+v2*v2;
 	}
 	*n0 = v1*sqrt(-2.*log(rsq)/rsq);
-- 
1.8.3.1

