From 81858892a0e49d3f81359b5e891b5781b41097cf Mon Sep 17 00:00:00 2001
From: Jo Bovy <bovy@astro.utoronto.ca>
Date: Tue, 26 Jul 2022 10:59:17 -0400
Subject: [PATCH 3/3] Just link to librebound

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index bb784db..42c5b6d 100644
--- a/setup.py
+++ b/setup.py
@@ -76,7 +76,7 @@ libreboundxmodule = Extension('libreboundx',
                     include_dirs = ['src'],
                     library_dirs = [],
                     runtime_library_dirs = ["."],
-                    libraries=['rebound'+suffix[:suffix.rfind('.')]],
+                    libraries=['rebound'],
                     define_macros=[ ('LIBREBOUNDX', None) ],
                     extra_compile_args=['-fstrict-aliasing', '-O3','-std=c99', '-fPIC', '-Wpointer-arith', ghash_arg],
                     extra_link_args=extra_link_args,
-- 
1.8.3.1

