diff --git a/setup.py b/setup.py
index 5855b27d78..6c86b6dedf 100644
--- a/setup.py
+++ b/setup.py
@@ -122,8 +122,10 @@ if platform.system() == 'Windows':
     extensions = ['parquet', 'icu', 'fts', 'tpch', 'json']
 
 is_android = hasattr(sys, 'getandroidapilevel')
+is_pyodide = 'PYODIDE' in os.environ
 use_jemalloc = (
     not is_android
+    and not is_pyodide
     and platform.system() == 'Linux'
     and platform.architecture()[0] == '64bit'
     and platform.machine() == 'x86_64'
@@ -184,12 +186,20 @@ if 'DUCKDB_LIBS' in os.environ:
 
 define_macros = [('DUCKDB_PYTHON_LIB_NAME', lib_name)]
 
+custom_platform = os.environ.get('DUCKDB_CUSTOM_PLATFORM')
+if custom_platform is not None:
+    define_macros.append(('DUCKDB_CUSTOM_PLATFORM', custom_platform))
+
 if platform.system() == 'Darwin':
     toolchain_args.extend(['-stdlib=libc++', '-mmacosx-version-min=10.7'])
 
 if platform.system() == 'Windows':
     define_macros.extend([('DUCKDB_BUILD_LIBRARY', None), ('WIN32', None)])
 
+if is_pyodide:
+    # show more useful error messages in the browser
+    define_macros.append(('PYBIND11_DETAILED_ERROR_MESSAGES', None))
+
 if 'BUILD_HTTPFS' in os.environ:
     libraries += ['crypto', 'ssl']
     extensions += ['httpfs']
