From 12917e4c1b9d64f3896989d767a4cec6a15684d6 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Thu, 31 Mar 2022 17:25:44 -0700
Subject: [PATCH 12/14] use local blas lapack

---
 numpy/distutils/system_info.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/numpy/distutils/system_info.py b/numpy/distutils/system_info.py
index d5a1687da..1a20b59c7 100644
--- a/numpy/distutils/system_info.py
+++ b/numpy/distutils/system_info.py
@@ -553,11 +553,11 @@ def get_info(name, notfound_action=0):
           'numeric': Numeric_info,
           'numarray': numarray_info,
           'numerix': numerix_info,
-          'lapack_opt': lapack_opt_info,
+          'lapack_opt': lapack_src_info,
           'lapack_ilp64_opt': lapack_ilp64_opt_info,
           'lapack_ilp64_plain_opt': lapack_ilp64_plain_opt_info,
           'lapack64__opt': lapack64__opt_info,
-          'blas_opt': blas_opt_info,
+          'blas_opt': blas_src_info,
           'blas_ilp64_opt': blas_ilp64_opt_info,
           'blas_ilp64_plain_opt': blas_ilp64_plain_opt_info,
           'blas64__opt': blas64__opt_info,
-- 
2.25.1

