diff --git a/numpy/linalg/umath_linalg.c.src b/numpy/linalg/umath_linalg.c.src
index 36b99b522..7c2d21189 100644
--- a/numpy/linalg/umath_linalg.c.src
+++ b/numpy/linalg/umath_linalg.c.src
@@ -797,7 +797,6 @@ update_pointers(npy_uint8** bases, ptrdiff_t* offsets, size_t count)
    positives with this warning
 */
 #pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"

 /*
  *****************************************************************************
