diff --git a/numpy/core/src/multiarray/nditer_pywrap.c b/numpy/core/src/multiarray/nditer_pywrap.c
index 0d318178f..2d3de0247 100644
--- a/numpy/core/src/multiarray/nditer_pywrap.c
+++ b/numpy/core/src/multiarray/nditer_pywrap.c
@@ -2481,7 +2481,7 @@ NPY_NO_EXPORT PyTypeObject NpyIter_Type = {
     0,                                          /* tp_iter */
     (iternextfunc)npyiter_next,                 /* tp_iternext */
     npyiter_methods,                            /* tp_methods */
-    npyiter_members,                            /* tp_members */
+    0,                                          /* tp_members */
     npyiter_getsets,                            /* tp_getset */
     0,                                          /* tp_base */
     0,                                          /* tp_dict */
