diff --git a/numpy/core/src/multiarray/nditer_pywrap.c b/numpy/core/src/multiarray/nditer_pywrap.c
index 8e072d5f4..897950c7c 100644
--- a/numpy/core/src/multiarray/nditer_pywrap.c
+++ b/numpy/core/src/multiarray/nditer_pywrap.c
@@ -2477,7 +2477,7 @@ NPY_NO_EXPORT PyTypeObject NpyIter_Type = {
     .tp_flags = Py_TPFLAGS_DEFAULT,
     .tp_iternext = (iternextfunc)npyiter_next,
     .tp_methods = npyiter_methods,
-    .tp_members = npyiter_members,
+    .tp_members = 0,
     .tp_getset = npyiter_getsets,
     .tp_init = (initproc)npyiter_init,
     .tp_new = npyiter_new,
