diff --git a/numpy/core/src/npymath/npy_math_complex.c.src b/numpy/core/src/npymath/npy_math_complex.c.src
index cf427dad8..6c9f1c37e 100644
--- a/numpy/core/src/npymath/npy_math_complex.c.src
+++ b/numpy/core/src/npymath/npy_math_complex.c.src
@@ -1522,10 +1522,9 @@ _sum_squares@c@(@type@ x, @type@ y)
 #if @precision@ == 1
 const npy_float SQRT_MIN = 1.0842022e-19f;
 #endif
-#if @precision@ == 2
+#if @precision@ == 2 || NPY_SIZEOF_LONGDOUBLE < 16
 const npy_double SQRT_MIN = 1.4916681462400413e-154; /* sqrt(DBL_MIN) */
-#endif
-#if @precision@ == 3
+#elif @precision@ == 3
 #if NPY_SIZEOF_LONGDOUBLE == NPY_SIZEOF_DOUBLE
 const npy_longdouble SQRT_MIN = 1.4916681462400413e-154; /* sqrt(DBL_MIN) */
 #else
