diff --git a/numpy/fft/fftpack_litemodule.c b/numpy/fft/fftpack_litemodule.c
index dfa0d211b..0bdf709c5 100644
--- a/numpy/fft/fftpack_litemodule.c
+++ b/numpy/fft/fftpack_litemodule.c
@@ -328,6 +328,8 @@ static struct PyModuleDef moduledef = {
 };
 #endif
 
+static volatile int *_dummy_malloc;
+
 /* Initialization function for the module */
 #if PY_MAJOR_VERSION >= 3
 #define RETVAL m
@@ -349,6 +351,10 @@ initfftpack_lite(void)
             (PyObject*)NULL,PYTHON_API_VERSION);
 #endif
 
+    _dummy_malloc = malloc(sizeof(int));
+    *_dummy_malloc = 1;
+    free((void *)_dummy_malloc);
+
     /* Import the array object */
     import_array();
 
