diff --git a/numpy/distutils/command/build.py b/numpy/distutils/command/build.py
index a4fda537d..a1c4bfdb1 100644
--- a/numpy/distutils/command/build.py
+++ b/numpy/distutils/command/build.py
@@ -35,8 +35,8 @@ def initialize_options(self):
         old_build.initialize_options(self)
         self.fcompiler = None
         self.warn_error = False
-        self.cpu_baseline = "min"
-        self.cpu_dispatch = "max -xop -fma4" # drop AMD legacy features by default
+        self.cpu_baseline = "avx"
+        self.cpu_dispatch = "avx" # drop AMD legacy features by default
         self.disable_optimization = False
         """
         the '_simd' module is a very large. Adding more dispatched features
