diff --git a/numpy/distutils/system_info.py b/numpy/distutils/system_info.py
index 7f41bb07e..c12d9b196 100644
--- a/numpy/distutils/system_info.py
+++ b/numpy/distutils/system_info.py
@@ -549,11 +549,11 @@ def get_info(name, notfound_action=0):
           'numeric': Numeric_info,
           'numarray': numarray_info,
           'numerix': numerix_info,
-          'lapack_opt': lapack_opt_info,
+          'lapack_opt': lapack_src_info,
           'lapack_ilp64_opt': lapack_ilp64_opt_info,
           'lapack_ilp64_plain_opt': lapack_ilp64_plain_opt_info,
           'lapack64__opt': lapack64__opt_info,
-          'blas_opt': blas_opt_info,
+          'blas_opt': blas_src_info,
           'blas_ilp64_opt': blas_ilp64_opt_info,
           'blas_ilp64_plain_opt': blas_ilp64_plain_opt_info,
           'blas64__opt': blas64__opt_info,
