diff --git a/numpy/fft/helper.py b/numpy/fft/helper.py
index 0856d6759..7238b74a2 100644
--- a/numpy/fft/helper.py
+++ b/numpy/fft/helper.py
@@ -5,7 +5,10 @@ Discrete Fourier Transforms - helper.py
 from __future__ import division, absolute_import, print_function

 import collections
-import threading
+try:
+    import threading
+except ImportError:
+    import dummy_threading as threading

 from numpy.compat import integer_types
 from numpy.core import (
