From 43c97d5a14e0ae8d8ccef6a4f1b6142081e7ac82 Mon Sep 17 00:00:00 2001
From: Julius Michaelis <gitter@liftm.de>
Date: Tue, 24 Mar 2020 23:17:16 +0900
Subject: [PATCH] pyodide: no mmap

---
 jplephem/daf.py | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/jplephem/daf.py b/jplephem/daf.py
index ce75875..f740422 100644
--- a/jplephem/daf.py
+++ b/jplephem/daf.py
@@ -102,16 +102,13 @@ class DAF(object):
         i, j = 8 * start - 8, 8 * end
         try:
             fileno = self.file.fileno()
-        except (AttributeError, io.UnsupportedOperation):
-            fileno = None
-        if fileno is None:
+            skip = i % mmap.ALLOCATIONGRANULARITY
+            r = mmap.ACCESS_READ
+            m = mmap.mmap(fileno, length=j-i+skip, access=r, offset=i-skip) # canary
+        except (AttributeError, io.UnsupportedOperation, OSError):
             skip = 0
             self.file.seek(i)
             m = self.file.read(j - i)
-        else:
-            skip = i % mmap.ALLOCATIONGRANULARITY
-            r = mmap.ACCESS_READ
-            m = mmap.mmap(fileno, length=j-i+skip, access=r, offset=i-skip)
         if sys.version_info > (3,):
             m = memoryview(m)  # so further slicing can return views
         return m, skip
-- 
2.25.0

