From bf250e6b263f8c4fddc1eb819ad5de85776b84f0 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Mon, 23 Jan 2023 19:29:57 -0800
Subject: [PATCH] Don't check libsodium

---
 setup.py | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/setup.py b/setup.py
index 505169b..de2fe67 100644
--- a/setup.py
+++ b/setup.py
@@ -168,9 +168,6 @@ class build_clib(_build_clib):
         # Build the library
         subprocess.check_call(["make"] + make_args, cwd=build_temp)
 
-        # Check the build library
-        subprocess.check_call(["make", "check"] + make_args, cwd=build_temp)
-
         # Install the built library
         subprocess.check_call(["make", "install"] + make_args, cwd=build_temp)
 
-- 
2.25.1

