From 56a32407ca20c0fc2227a66640ff3eded4fd22c9 Mon Sep 17 00:00:00 2001
From: Agriya Khetarpal <74401230+agriyakhetarpal@users.noreply.github.com>
Date: Tue, 10 Dec 2024 12:34:54 +0530
Subject: [PATCH 3/5] Temporarily disable event listeners, add notes

---
 matplotlib_pyodide/browser_backend.py | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/matplotlib_pyodide/browser_backend.py b/matplotlib_pyodide/browser_backend.py
index 130c6f1..be24a5a 100644
--- a/matplotlib_pyodide/browser_backend.py
+++ b/matplotlib_pyodide/browser_backend.py
@@ -164,13 +164,15 @@ class FigureCanvasWasm(FigureCanvasBase):
         rubberband.setAttribute("tabindex", "0")
         # Event handlers are added to the canvas "on top", even though most of
         # the activity happens in the canvas below.
-        add_event_listener(rubberband, "mousemove", self.onmousemove)
-        add_event_listener(rubberband, "mouseup", self.onmouseup)
-        add_event_listener(rubberband, "mousedown", self.onmousedown)
-        add_event_listener(rubberband, "mouseenter", self.onmouseenter)
-        add_event_listener(rubberband, "mouseleave", self.onmouseleave)
-        add_event_listener(rubberband, "keyup", self.onkeyup)
-        add_event_listener(rubberband, "keydown", self.onkeydown)
+        # TODO: with 0.2.3, we temporarily disable event listeners for the rubberband canvas.
+        # This shall be revisited in a future release.
+        # add_event_listener(rubberband, "mousemove", self.onmousemove)
+        # add_event_listener(rubberband, "mouseup", self.onmouseup)
+        # add_event_listener(rubberband, "mousedown", self.onmousedown)
+        # add_event_listener(rubberband, "mouseenter", self.onmouseenter)
+        # add_event_listener(rubberband, "mouseleave", self.onmouseleave)
+        # add_event_listener(rubberband, "keyup", self.onkeyup)
+        # add_event_listener(rubberband, "keydown", self.onkeydown)
         context = rubberband.getContext("2d")
         context.strokeStyle = "#000000"
         context.setLineDash([2, 2])
-- 
2.39.5 (Apple Git-154)

