diff -ur matplotlib-2.2.2/lib/matplotlib/backends/backend_agg.py matplotlib-2.2.2/lib/matplotlib/backends/backend_agg.py
--- a/lib/matplotlib/backends/backend_agg.py	2018-03-17 14:03:23.000000000 -0400
+++ b/lib/matplotlib/backends/backend_agg.py	2018-05-17 09:43:34.710665159 -0400
@@ -24,7 +24,10 @@

 import six

-import threading
+try:
+    import threading
+except ImportError:
+    import dummy_threading as threading
 import numpy as np
 from collections import OrderedDict
 from math import radians, cos, sin
diff -ur matplotlib-2.2.2/lib/matplotlib/font_manager.py matplotlib-2.2.2/lib/matplotlib/font_manager.py
--- a/lib/matplotlib/font_manager.py	2018-03-17 14:03:23.000000000 -0400
+++ b/lib/matplotlib/font_manager.py	2018-05-16 16:56:49.399466649 -0400
@@ -48,7 +48,10 @@
 import json
 import os
 import sys
-from threading import Timer
+try:
+    from threading import Timer
+except ImportError:
+    from dummy_threading import Timer
 import warnings
 import logging

