Proper Freetype linking.

diff --git a/setupext.py b/setupext.py
index d0e26d484d..82829f7984 100644
--- a/setupext.py
+++ b/setupext.py
@@ -555,6 +555,9 @@ class FreeType(SetupPackage):
 
     @classmethod
     def add_flags(cls, ext):
+        if "PYODIDE" in os.environ:
+            ext.libraries.append('freetype')
+            return
         # checkdep_freetype2.c immediately aborts the compilation either with
         # "foo.h: No such file or directory" if the header is not found, or an
         # appropriate error message if the header indicates a too-old version.
-- 
2.35.1

